// v-code-diff/dist/v3/index.es.js
(function(){"use strict";try{if(typeof document!="undefined"){var o=document.createElement("style");o.appendChild(document.createTextNode(".code-diff-view{--color-canvas-default-transparent: rgba(255,255,255,0);--color-page-header-bg: #f6f8fa;--color-marketing-icon-primary: #218bff;--color-marketing-icon-secondary: #54aeff;--color-diff-blob-addition-num-text: #24292f;--color-diff-blob-addition-fg: #24292f;--color-diff-blob-addition-num-bg: #ccffd8;--color-diff-blob-addition-line-bg: #e6ffec;--color-diff-blob-addition-word-bg: #abf2bc;--color-diff-blob-deletion-num-text: #24292f;--color-diff-blob-deletion-fg: #24292f;--color-diff-blob-deletion-num-bg: #ffd7d5;--color-diff-blob-deletion-line-bg: #ffebe9;--color-diff-blob-deletion-word-bg: rgba(255,129,130,.4);--color-diff-blob-hunk-num-bg: rgba(84,174,255,.4);--color-diff-blob-expander-icon: #57606a;--color-diff-blob-selected-line-highlight-mix-blend-mode: multiply;--color-diffstat-deletion-border: rgba(27,31,36,.15);--color-diffstat-addition-border: rgba(27,31,36,.15);--color-diffstat-addition-bg: #2da44e;--color-search-keyword-hl: #fff8c5;--color-prettylights-syntax-comment: #6e7781;--color-prettylights-syntax-constant: #0550ae;--color-prettylights-syntax-entity: #8250df;--color-prettylights-syntax-storage-modifier-import: #24292f;--color-prettylights-syntax-entity-tag: #116329;--color-prettylights-syntax-keyword: #cf222e;--color-prettylights-syntax-string: #0a3069;--color-prettylights-syntax-variable: #953800;--color-prettylights-syntax-brackethighlighter-unmatched: #82071e;--color-prettylights-syntax-invalid-illegal-text: #f6f8fa;--color-prettylights-syntax-invalid-illegal-bg: #82071e;--color-prettylights-syntax-carriage-return-text: #f6f8fa;--color-prettylights-syntax-carriage-return-bg: #cf222e;--color-prettylights-syntax-string-regexp: #116329;--color-prettylights-syntax-markup-list: #3b2300;--color-prettylights-syntax-markup-heading: #0550ae;--color-prettylights-syntax-markup-italic: #24292f;--color-prettylights-syntax-markup-bold: #24292f;--color-prettylights-syntax-markup-deleted-text: #82071e;--color-prettylights-syntax-markup-deleted-bg: #ffebe9;--color-prettylights-syntax-markup-inserted-text: #116329;--color-prettylights-syntax-markup-inserted-bg: #dafbe1;--color-prettylights-syntax-markup-changed-text: #953800;--color-prettylights-syntax-markup-changed-bg: #ffd8b5;--color-prettylights-syntax-markup-ignored-text: #eaeef2;--color-prettylights-syntax-markup-ignored-bg: #0550ae;--color-prettylights-syntax-meta-diff-range: #8250df;--color-prettylights-syntax-brackethighlighter-angle: #57606a;--color-prettylights-syntax-sublimelinter-gutter-mark: #8c959f;--color-prettylights-syntax-constant-other-reference-link: #0a3069;--color-codemirror-text: #24292f;--color-codemirror-bg: #ffffff;--color-codemirror-gutters-bg: #ffffff;--color-codemirror-guttermarker-text: #ffffff;--color-codemirror-guttermarker-subtle-text: #6e7781;--color-codemirror-linenumber-text: #57606a;--color-codemirror-cursor: #24292f;--color-codemirror-selection-bg: rgba(84,174,255,.4);--color-codemirror-activeline-bg: rgba(234,238,242,.5);--color-codemirror-matchingbracket-text: #24292f;--color-codemirror-lines-bg: #ffffff;--color-codemirror-syntax-comment: #24292f;--color-codemirror-syntax-constant: #0550ae;--color-codemirror-syntax-entity: #8250df;--color-codemirror-syntax-keyword: #cf222e;--color-codemirror-syntax-storage: #cf222e;--color-codemirror-syntax-string: #0a3069;--color-codemirror-syntax-support: #0550ae;--color-codemirror-syntax-variable: #953800;--color-checks-bg: #24292f;--color-checks-run-border-width: 0px;--color-checks-container-border-width: 0px;--color-checks-text-primary: #f6f8fa;--color-checks-text-secondary: #8c959f;--color-checks-text-link: #54aeff;--color-checks-btn-icon: #afb8c1;--color-checks-btn-hover-icon: #f6f8fa;--color-checks-btn-hover-bg: rgba(255,255,255,.125);--color-checks-input-text: #eaeef2;--color-checks-input-placeholder-text: #8c959f;--color-checks-input-focus-text: #8c959f;--color-checks-input-bg: #32383f;--color-checks-input-shadow: none;--color-checks-donut-error: #fa4549;--color-checks-donut-pending: #bf8700;--color-checks-donut-success: #2da44e;--color-checks-donut-neutral: #afb8c1;--color-checks-dropdown-text: #afb8c1;--color-checks-dropdown-bg: #32383f;--color-checks-dropdown-border: #424a53;--color-checks-dropdown-shadow: rgba(27,31,36,.3);--color-checks-dropdown-hover-text: #f6f8fa;--color-checks-dropdown-hover-bg: #424a53;--color-checks-dropdown-btn-hover-text: #f6f8fa;--color-checks-dropdown-btn-hover-bg: #32383f;--color-checks-scrollbar-thumb-bg: #57606a;--color-checks-header-label-text: #d0d7de;--color-checks-header-label-open-text: #f6f8fa;--color-checks-header-border: #32383f;--color-checks-header-icon: #8c959f;--color-checks-line-text: #d0d7de;--color-checks-line-num-text: rgba(140,149,159,.75);--color-checks-line-timestamp-text: #8c959f;--color-checks-line-hover-bg: #32383f;--color-checks-line-selected-bg: rgba(33,139,255,.15);--color-checks-line-selected-num-text: #54aeff;--color-checks-line-dt-fm-text: #24292f;--color-checks-line-dt-fm-bg: #9a6700;--color-checks-gate-bg: rgba(125,78,0,.15);--color-checks-gate-text: #d0d7de;--color-checks-gate-waiting-text: #d4a72c;--color-checks-step-header-open-bg: #32383f;--color-checks-step-error-text: #ff8182;--color-checks-step-warning-text: #d4a72c;--color-checks-logline-text: #8c959f;--color-checks-logline-num-text: rgba(140,149,159,.75);--color-checks-logline-debug-text: #c297ff;--color-checks-logline-error-text: #d0d7de;--color-checks-logline-error-num-text: #ff8182;--color-checks-logline-error-bg: rgba(164,14,38,.15);--color-checks-logline-warning-text: #d0d7de;--color-checks-logline-warning-num-text: #d4a72c;--color-checks-logline-warning-bg: rgba(125,78,0,.15);--color-checks-logline-command-text: #54aeff;--color-checks-logline-section-text: #4ac26b;--color-checks-ansi-black: #24292f;--color-checks-ansi-black-bright: #32383f;--color-checks-ansi-white: #d0d7de;--color-checks-ansi-white-bright: #d0d7de;--color-checks-ansi-gray: #8c959f;--color-checks-ansi-red: #ff8182;--color-checks-ansi-red-bright: #ffaba8;--color-checks-ansi-green: #4ac26b;--color-checks-ansi-green-bright: #6fdd8b;--color-checks-ansi-yellow: #d4a72c;--color-checks-ansi-yellow-bright: #eac54f;--color-checks-ansi-blue: #54aeff;--color-checks-ansi-blue-bright: #80ccff;--color-checks-ansi-magenta: #c297ff;--color-checks-ansi-magenta-bright: #d8b9ff;--color-checks-ansi-cyan: #76e3ea;--color-checks-ansi-cyan-bright: #b3f0ff;--color-project-header-bg: #24292f;--color-project-sidebar-bg: #ffffff;--color-project-gradient-in: #ffffff;--color-project-gradient-out: rgba(255,255,255,0);--color-mktg-btn-bg: #1b1f23;--color-mktg-btn-shadow-outline: rgb(0 0 0 / 15%) 0 0 0 1px inset;--color-mktg-btn-shadow-focus: rgb(0 0 0 / 15%) 0 0 0 4px;--color-mktg-btn-shadow-hover: 0 3px 2px rgba(0, 0, 0, .07), 0 7px 5px rgba(0, 0, 0, .04), 0 12px 10px rgba(0, 0, 0, .03), 0 22px 18px rgba(0, 0, 0, .03), 0 42px 33px rgba(0, 0, 0, .02), 0 100px 80px rgba(0, 0, 0, .02);--color-mktg-btn-shadow-hover-muted: rgb(0 0 0 / 70%) 0 0 0 2px inset;--color-avatar-bg: #ffffff;--color-avatar-border: rgba(27,31,36,.15);--color-avatar-stack-fade: #afb8c1;--color-avatar-stack-fade-more: #d0d7de;--color-avatar-child-shadow: -2px -2px 0 rgba(255,255,255,.8);--color-topic-tag-border: rgba(0,0,0,0);--color-counter-border: rgba(0,0,0,0);--color-select-menu-backdrop-border: rgba(0,0,0,0);--color-select-menu-tap-highlight: rgba(175,184,193,.5);--color-select-menu-tap-focus-bg: #b6e3ff;--color-overlay-shadow: 0 1px 3px rgba(27,31,36,.12), 0 8px 24px rgba(66,74,83,.12);--color-header-text: rgba(255,255,255,.7);--color-header-bg: #24292f;--color-header-divider: #57606a;--color-header-logo: #ffffff;--color-header-search-bg: #24292f;--color-header-search-border: #57606a;--color-sidenav-selected-bg: #ffffff;--color-menu-bg-active: rgba(0,0,0,0);--color-input-disabled-bg: rgba(175,184,193,.2);--color-timeline-badge-bg: #eaeef2;--color-ansi-black: #24292f;--color-ansi-black-bright: #57606a;--color-ansi-white: #6e7781;--color-ansi-white-bright: #8c959f;--color-ansi-gray: #6e7781;--color-ansi-red: #cf222e;--color-ansi-red-bright: #a40e26;--color-ansi-green: #116329;--color-ansi-green-bright: #1a7f37;--color-ansi-yellow: #4d2d00;--color-ansi-yellow-bright: #633c01;--color-ansi-blue: #0969da;--color-ansi-blue-bright: #218bff;--color-ansi-magenta: #8250df;--color-ansi-magenta-bright: #a475f9;--color-ansi-cyan: #1b7c83;--color-ansi-cyan-bright: #3192aa;--color-btn-text: #24292f;--color-btn-bg: #f6f8fa;--color-btn-border: rgba(27,31,36,.15);--color-btn-shadow: 0 1px 0 rgba(27,31,36,.04);--color-btn-inset-shadow: inset 0 1px 0 rgba(255,255,255,.25);--color-btn-hover-bg: #f3f4f6;--color-btn-hover-border: rgba(27,31,36,.15);--color-btn-active-bg: hsla(220,14%,93%,1);--color-btn-active-border: rgba(27,31,36,.15);--color-btn-selected-bg: hsla(220,14%,94%,1);--color-btn-focus-bg: #f6f8fa;--color-btn-focus-border: rgba(27,31,36,.15);--color-btn-focus-shadow: 0 0 0 3px rgba(9,105,218,.3);--color-btn-shadow-active: inset 0 .15em .3em rgba(27,31,36,.15);--color-btn-shadow-input-focus: 0 0 0 .2em rgba(9,105,218,.3);--color-btn-counter-bg: rgba(27,31,36,.08);--color-btn-primary-text: #ffffff;--color-btn-primary-bg: #2da44e;--color-btn-primary-border: rgba(27,31,36,.15);--color-btn-primary-shadow: 0 1px 0 rgba(27,31,36,.1);--color-btn-primary-inset-shadow: inset 0 1px 0 rgba(255,255,255,.03);--color-btn-primary-hover-bg: #2c974b;--color-btn-primary-hover-border: rgba(27,31,36,.15);--color-btn-primary-selected-bg: hsla(137,55%,36%,1);--color-btn-primary-selected-shadow: inset 0 1px 0 rgba(0,45,17,.2);--color-btn-primary-disabled-text: rgba(255,255,255,.8);--color-btn-primary-disabled-bg: #94d3a2;--color-btn-primary-disabled-border: rgba(27,31,36,.15);--color-btn-primary-focus-bg: #2da44e;--color-btn-primary-focus-border: rgba(27,31,36,.15);--color-btn-primary-focus-shadow: 0 0 0 3px rgba(45,164,78,.4);--color-btn-primary-icon: rgba(255,255,255,.8);--color-btn-primary-counter-bg: rgba(255,255,255,.2);--color-btn-outline-text: #0969da;--color-btn-outline-hover-text: #ffffff;--color-btn-outline-hover-bg: #0969da;--color-btn-outline-hover-border: rgba(27,31,36,.15);--color-btn-outline-hover-shadow: 0 1px 0 rgba(27,31,36,.1);--color-btn-outline-hover-inset-shadow: inset 0 1px 0 rgba(255,255,255,.03);--color-btn-outline-hover-counter-bg: rgba(255,255,255,.2);--color-btn-outline-selected-text: #ffffff;--color-btn-outline-selected-bg: hsla(212,92%,42%,1);--color-btn-outline-selected-border: rgba(27,31,36,.15);--color-btn-outline-selected-shadow: inset 0 1px 0 rgba(0,33,85,.2);--color-btn-outline-disabled-text: rgba(9,105,218,.5);--color-btn-outline-disabled-bg: #f6f8fa;--color-btn-outline-disabled-counter-bg: rgba(9,105,218,.05);--color-btn-outline-focus-border: rgba(27,31,36,.15);--color-btn-outline-focus-shadow: 0 0 0 3px rgba(5,80,174,.4);--color-btn-outline-counter-bg: rgba(9,105,218,.1);--color-btn-danger-text: #cf222e;--color-btn-danger-hover-text: #ffffff;--color-btn-danger-hover-bg: #a40e26;--color-btn-danger-hover-border: rgba(27,31,36,.15);--color-btn-danger-hover-shadow: 0 1px 0 rgba(27,31,36,.1);--color-btn-danger-hover-inset-shadow: inset 0 1px 0 rgba(255,255,255,.03);--color-btn-danger-hover-counter-bg: rgba(255,255,255,.2);--color-btn-danger-selected-text: #ffffff;--color-btn-danger-selected-bg: hsla(356,72%,44%,1);--color-btn-danger-selected-border: rgba(27,31,36,.15);--color-btn-danger-selected-shadow: inset 0 1px 0 rgba(76,0,20,.2);--color-btn-danger-disabled-text: rgba(207,34,46,.5);--color-btn-danger-disabled-bg: #f6f8fa;--color-btn-danger-disabled-counter-bg: rgba(207,34,46,.05);--color-btn-danger-focus-border: rgba(27,31,36,.15);--color-btn-danger-focus-shadow: 0 0 0 3px rgba(164,14,38,.4);--color-btn-danger-counter-bg: rgba(207,34,46,.1);--color-btn-danger-icon: #cf222e;--color-btn-danger-hover-icon: #ffffff;--color-underlinenav-icon: #6e7781;--color-underlinenav-border-hover: rgba(175,184,193,.2);--color-action-list-item-inline-divider: rgba(208,215,222,.48);--color-action-list-item-default-hover-bg: rgba(208,215,222,.32);--color-action-list-item-default-hover-border: rgba(0,0,0,0);--color-action-list-item-default-active-bg: rgba(208,215,222,.48);--color-action-list-item-default-active-border: rgba(0,0,0,0);--color-action-list-item-default-selected-bg: rgba(208,215,222,.24);--color-action-list-item-danger-hover-bg: rgba(255,235,233,.64);--color-action-list-item-danger-active-bg: #ffebe9;--color-action-list-item-danger-hover-text: #cf222e;--color-switch-track-bg: #eaeef2;--color-switch-track-border: #afb8c1;--color-switch-track-checked-bg: #ddf4ff;--color-switch-track-checked-hover-bg: #b6e3ff;--color-switch-track-checked-active-bg: #80ccff;--color-switch-track-checked-border: #54aeff;--color-switch-knob-checked-bg: #0969da;--color-switch-knob-checked-disabled-bg: #6e7781;--color-segmented-control-bg: #eaeef2;--color-segmented-control-button-hover-bg: rgba(175,184,193,.2);--color-segmented-control-button-active-bg: rgba(175,184,193,.4);--color-segmented-control-button-selected-border: #6e7781;--color-fg-default: #24292f;--color-fg-muted: #57606a;--color-fg-subtle: #6e7781;--color-fg-on-emphasis: #ffffff;--color-canvas-default: #ffffff;--color-canvas-overlay: #ffffff;--color-canvas-inset: #f6f8fa;--color-canvas-subtle: #f6f8fa;--color-border-default: #d0d7de;--color-border-muted: hsla(210,18%,87%,1);--color-border-subtle: rgba(27,31,36,.15);--color-shadow-small: 0 1px 0 rgba(27,31,36,.04);--color-shadow-medium: 0 3px 6px rgba(140,149,159,.15);--color-shadow-large: 0 8px 24px rgba(140,149,159,.2);--color-shadow-extra-large: 0 12px 28px rgba(140,149,159,.3);--color-neutral-emphasis-plus: #24292f;--color-neutral-emphasis: #6e7781;--color-neutral-muted: rgba(175,184,193,.2);--color-neutral-subtle: rgba(234,238,242,.5);--color-accent-fg: #0969da;--color-accent-emphasis: #0969da;--color-accent-muted: rgba(84,174,255,.4);--color-accent-subtle: #ddf4ff;--color-success-fg: #1a7f37;--color-success-emphasis: #2da44e;--color-success-muted: rgba(74,194,107,.4);--color-success-subtle: #dafbe1;--color-attention-fg: #9a6700;--color-attention-emphasis: #bf8700;--color-attention-muted: rgba(212,167,44,.4);--color-attention-subtle: #fff8c5;--color-severe-fg: #bc4c00;--color-severe-emphasis: #bc4c00;--color-severe-muted: rgba(251,143,68,.4);--color-severe-subtle: #fff1e5;--color-danger-fg: #cf222e;--color-danger-emphasis: #cf222e;--color-danger-muted: rgba(255,129,130,.4);--color-danger-subtle: #ffebe9;--color-open-fg: #1a7f37;--color-open-emphasis: #2da44e;--color-open-muted: rgba(74,194,107,.4);--color-open-subtle: #dafbe1;--color-closed-fg: #cf222e;--color-closed-emphasis: #cf222e;--color-closed-muted: rgba(255,129,130,.4);--color-closed-subtle: #ffebe9;--color-done-fg: #8250df;--color-done-emphasis: #8250df;--color-done-muted: rgba(194,151,255,.4);--color-done-subtle: #fbefff;--color-sponsors-fg: #bf3989;--color-sponsors-emphasis: #bf3989;--color-sponsors-muted: rgba(255,128,200,.4);--color-sponsors-subtle: #ffeff7;--color-primer-fg-disabled: #8c959f;--color-primer-canvas-backdrop: rgba(27,31,36,.5);--color-primer-canvas-sticky: rgba(255,255,255,.95);--color-primer-border-active: #fd8c73;--color-primer-border-contrast: rgba(27,31,36,.1);--color-primer-shadow-highlight: inset 0 1px 0 rgba(255,255,255,.25);--color-primer-shadow-inset: inset 0 1px 0 rgba(208,215,222,.2);--color-primer-shadow-focus: 0 0 0 3px rgba(9,105,218,.3);--color-scale-black: #1b1f24;--color-scale-white: #ffffff;--color-scale-gray-0: #f6f8fa;--color-scale-gray-1: #eaeef2;--color-scale-gray-2: #d0d7de;--color-scale-gray-3: #afb8c1;--color-scale-gray-4: #8c959f;--color-scale-gray-5: #6e7781;--color-scale-gray-6: #57606a;--color-scale-gray-7: #424a53;--color-scale-gray-8: #32383f;--color-scale-gray-9: #24292f;--color-scale-blue-0: #ddf4ff;--color-scale-blue-1: #b6e3ff;--color-scale-blue-2: #80ccff;--color-scale-blue-3: #54aeff;--color-scale-blue-4: #218bff;--color-scale-blue-5: #0969da;--color-scale-blue-6: #0550ae;--color-scale-blue-7: #033d8b;--color-scale-blue-8: #0a3069;--color-scale-blue-9: #002155;--color-scale-green-0: #dafbe1;--color-scale-green-1: #aceebb;--color-scale-green-2: #6fdd8b;--color-scale-green-3: #4ac26b;--color-scale-green-4: #2da44e;--color-scale-green-5: #1a7f37;--color-scale-green-6: #116329;--color-scale-green-7: #044f1e;--color-scale-green-8: #003d16;--color-scale-green-9: #002d11;--color-scale-yellow-0: #fff8c5;--color-scale-yellow-1: #fae17d;--color-scale-yellow-2: #eac54f;--color-scale-yellow-3: #d4a72c;--color-scale-yellow-4: #bf8700;--color-scale-yellow-5: #9a6700;--color-scale-yellow-6: #7d4e00;--color-scale-yellow-7: #633c01;--color-scale-yellow-8: #4d2d00;--color-scale-yellow-9: #3b2300;--color-scale-orange-0: #fff1e5;--color-scale-orange-1: #ffd8b5;--color-scale-orange-2: #ffb77c;--color-scale-orange-3: #fb8f44;--color-scale-orange-4: #e16f24;--color-scale-orange-5: #bc4c00;--color-scale-orange-6: #953800;--color-scale-orange-7: #762c00;--color-scale-orange-8: #5c2200;--color-scale-orange-9: #471700;--color-scale-red-0: #ffebe9;--color-scale-red-1: #ffcecb;--color-scale-red-2: #ffaba8;--color-scale-red-3: #ff8182;--color-scale-red-4: #fa4549;--color-scale-red-5: #cf222e;--color-scale-red-6: #a40e26;--color-scale-red-7: #82071e;--color-scale-red-8: #660018;--color-scale-red-9: #4c0014;--color-scale-purple-0: #fbefff;--color-scale-purple-1: #ecd8ff;--color-scale-purple-2: #d8b9ff;--color-scale-purple-3: #c297ff;--color-scale-purple-4: #a475f9;--color-scale-purple-5: #8250df;--color-scale-purple-6: #6639ba;--color-scale-purple-7: #512a97;--color-scale-purple-8: #3e1f79;--color-scale-purple-9: #2e1461;--color-scale-pink-0: #ffeff7;--color-scale-pink-1: #ffd3eb;--color-scale-pink-2: #ffadda;--color-scale-pink-3: #ff80c8;--color-scale-pink-4: #e85aad;--color-scale-pink-5: #bf3989;--color-scale-pink-6: #99286e;--color-scale-pink-7: #772057;--color-scale-pink-8: #611347;--color-scale-pink-9: #4d0336;--color-scale-coral-0: #fff0eb;--color-scale-coral-1: #ffd6cc;--color-scale-coral-2: #ffb4a1;--color-scale-coral-3: #fd8c73;--color-scale-coral-4: #ec6547;--color-scale-coral-5: #c4432b;--color-scale-coral-6: #9e2f1c;--color-scale-coral-7: #801f0f;--color-scale-coral-8: #691105;--color-scale-coral-9: #510901}.code-diff-view{position:relative;margin-top:16px;margin-bottom:16px;border:1px solid var(--color-border-default, #ddd);border-radius:6px;overflow-y:auto}.code-diff-view *{position:static;box-sizing:border-box}.code-diff-view .file-header{background-color:var(--color-canvas-subtle);border-bottom:1px solid var(--color-border-default);padding:8px 16px;font-size:12px;font-family:ui-monospace,SFMono-Regular,SF Mono,Menlo,Consolas,Liberation Mono,monospace}.code-diff-view .file-header .file-info{display:flex;justify-content:space-between;align-items:center;margin-left:8px;height:24px}.code-diff-view .file-header .file-info .filename{font-size:13px;color:var(--color-fg-default)}.code-diff-view .file-header .file-info .diff-stat .diff-stat-added{color:var(--color-diffstat-addition-bg)}.code-diff-view .file-header .file-info .diff-stat .diff-stat-deleted{color:var(--color-danger-emphasis)}.code-diff-view table{border-spacing:0}.code-diff-view .diff-table{width:100%}.code-diff-view .diff-table .blob-num{position:relative;width:1%;min-width:50px;padding-right:10px;padding-left:10px;font-family:ui-monospace,SFMono-Regular,SF Mono,Menlo,Consolas,Liberation Mono,monospace;font-size:12px;line-height:20px;color:var(--color-fg-subtle);text-align:right;white-space:nowrap;vertical-align:top;cursor:pointer;-webkit-user-select:none;user-select:none}.code-diff-view .diff-table .blob-num-deletion{color:var(--color-diff-blob-deletion-num-text);background-color:var(--color-diff-blob-deletion-num-bg);border-color:var(--color-danger-emphasis)}.code-diff-view .diff-table .blob-num-addition{color:var(--color-diff-blob-addition-num-text);background-color:var(--color-diff-blob-addition-num-bg);border-color:var(--color-success-emphasis)}.code-diff-view .diff-table .blob-code{position:relative;padding-right:10px;padding-left:10px;line-height:20px;vertical-align:top}.code-diff-view .diff-table .blob-code .blob-code-inner{display:table-cell;overflow:visible;font-family:ui-monospace,SFMono-Regular,SF Mono,Menlo,Consolas,Liberation Mono,monospace;font-size:12px;color:var(--color-fg-default);word-wrap:anywhere;white-space:pre-wrap}.code-diff-view .diff-table .blob-code-deletion{background-color:var(--color-diff-blob-deletion-line-bg);outline:1px dashed transparent}.code-diff-view .diff-table .blob-code-deletion .x{color:var(--color-diff-blob-deletion-fg);background-color:var(--color-diff-blob-deletion-word-bg)}.code-diff-view .diff-table .blob-code-addition{background-color:var(--color-diff-blob-addition-line-bg);outline:1px dotted transparent}.code-diff-view .diff-table .blob-code-addition .x{color:var(--color-diff-blob-addition-fg);background-color:var(--color-diff-blob-addition-word-bg)}.code-diff-view .diff-table .blob-code-context,.code-diff-view .diff-table .blob-code-addition,.code-diff-view .diff-table .blob-code-deletion{padding-left:22px!important}.code-diff-view .diff-table .blob-code-marker:before{position:absolute;top:1px;left:8px;padding-right:8px;content:attr(data-code-marker)}.code-diff-view .diff-table .blob-num-hunk{background-color:var(--color-diff-blob-hunk-num-bg)}.code-diff-view .diff-table .blob-code-hunk{background-color:var(--color-accent-subtle)}.code-diff-view .file-diff-split{table-layout:fixed}.code-diff-view .file-diff-split .blob-code+.blob-num{border-left:1px solid var(--color-border-muted)}.code-diff-view .empty-cell{cursor:default;background-color:var(--color-neutral-subtle);border-right-color:var(--color-border-muted)}pre code.hljs{display:block;overflow-x:auto;padding:1em}code.hljs{padding:3px 5px}.hljs{background:#fff;color:#000}.hljs-comment,.hljs-quote,.hljs-variable{color:green}.hljs-built_in,.hljs-keyword,.hljs-name,.hljs-selector-tag,.hljs-tag{color:#00f}.hljs-addition,.hljs-attribute,.hljs-literal,.hljs-section,.hljs-string,.hljs-template-tag,.hljs-template-variable,.hljs-title,.hljs-type{color:#a31515}.hljs-deletion,.hljs-meta,.hljs-selector-attr,.hljs-selector-pseudo{color:#2b91af}.hljs-doctag{color:gray}.hljs-attr{color:red}.hljs-bullet,.hljs-link,.hljs-symbol{color:#00b0e8}.hljs-emphasis{font-style:italic}.hljs-strong{font-weight:700}")),document.head.appendChild(o)}}catch(e){console.error("vite-plugin-css-injected-by-js",e)}})();
import { defineComponent as ye, openBlock as W, createElementBlock as j, createElementVNode as P, normalizeClass as be, unref as z, toDisplayString as he, createCommentVNode as fn, Fragment as ge, renderList as Fe, createBlock as Oe, computed as pe, ref as Un, watch as Pn, normalizeStyle as xn } from "vue";
function re() {
}
re.prototype = {
  diff: function(r, o) {
    var u = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : {}, h = u.callback;
    typeof u == "function" && (h = u, u = {}), this.options = u;
    var n = this;
    function t(p) {
      return h ? (setTimeout(function() {
        h(void 0, p);
      }, 0), !0) : p;
    }
    r = this.castInput(r), o = this.castInput(o), r = this.removeEmpty(this.tokenize(r)), o = this.removeEmpty(this.tokenize(o));
    var i = o.length, a = r.length, s = 1, l = i + a;
    u.maxEditLength && (l = Math.min(l, u.maxEditLength));
    var c = [{
      newPos: -1,
      components: []
    }], f = this.extractCommon(c[0], o, r, 0);
    if (c[0].newPos + 1 >= i && f + 1 >= a)
      return t([{
        value: this.join(o),
        count: o.length
      }]);
    function g() {
      for (var p = -1 * s; p <= s; p += 2) {
        var b = void 0, E = c[p - 1], v = c[p + 1], w = (v ? v.newPos : 0) - p;
        E && (c[p - 1] = void 0);
        var m = E && E.newPos + 1 < i, D = v && 0 <= w && w < a;
        if (!m && !D) {
          c[p] = void 0;
          continue;
        }
        if (!m || D && E.newPos < v.newPos ? (b = Fn(v), n.pushComponent(b.components, void 0, !0)) : (b = E, b.newPos++, n.pushComponent(b.components, !0, void 0)), w = n.extractCommon(b, o, r, p), b.newPos + 1 >= i && w + 1 >= a)
          return t(Hn(n, b.components, o, r, n.useLongestToken));
        c[p] = b;
      }
      s++;
    }
    if (h)
      (function p() {
        setTimeout(function() {
          if (s > l)
            return h();
          g() || p();
        }, 0);
      })();
    else
      for (; s <= l; ) {
        var d = g();
        if (d)
          return d;
      }
  },
  pushComponent: function(r, o, u) {
    var h = r[r.length - 1];
    h && h.added === o && h.removed === u ? r[r.length - 1] = {
      count: h.count + 1,
      added: o,
      removed: u
    } : r.push({
      count: 1,
      added: o,
      removed: u
    });
  },
  extractCommon: function(r, o, u, h) {
    for (var n = o.length, t = u.length, i = r.newPos, a = i - h, s = 0; i + 1 < n && a + 1 < t && this.equals(o[i + 1], u[a + 1]); )
      i++, a++, s++;
    return s && r.components.push({
      count: s
    }), r.newPos = i, a;
  },
  equals: function(r, o) {
    return this.options.comparator ? this.options.comparator(r, o) : r === o || this.options.ignoreCase && r.toLowerCase() === o.toLowerCase();
  },
  removeEmpty: function(r) {
    for (var o = [], u = 0; u < r.length; u++)
      r[u] && o.push(r[u]);
    return o;
  },
  castInput: function(r) {
    return r;
  },
  tokenize: function(r) {
    return r.split("");
  },
  join: function(r) {
    return r.join("");
  }
};
function Hn(e, r, o, u, h) {
  for (var n = 0, t = r.length, i = 0, a = 0; n < t; n++) {
    var s = r[n];
    if (s.removed) {
      if (s.value = e.join(u.slice(a, a + s.count)), a += s.count, n && r[n - 1].added) {
        var c = r[n - 1];
        r[n - 1] = r[n], r[n] = c;
      }
    } else {
      if (!s.added && h) {
        var l = o.slice(i, i + s.count);
        l = l.map(function(g, d) {
          var p = u[a + d];
          return p.length > g.length ? p : g;
        }), s.value = e.join(l);
      } else
        s.value = e.join(o.slice(i, i + s.count));
      i += s.count, s.added || (a += s.count);
    }
  }
  var f = r[t - 1];
  return t > 1 && typeof f.value == "string" && (f.added || f.removed) && e.equals("", f.value) && (r[t - 2].value += f.value, r.pop()), r;
}
function Fn(e) {
  return {
    newPos: e.newPos,
    components: e.components.slice(0)
  };
}
var zn = new re();
function Gn(e, r, o) {
  return zn.diff(e, r, o);
}
function Kn(e, r) {
  if (typeof e == "function")
    r.callback = e;
  else if (e)
    for (var o in e)
      e.hasOwnProperty(o) && (r[o] = e[o]);
  return r;
}
var qe = /^[A-Za-z\xC0-\u02C6\u02C8-\u02D7\u02DE-\u02FF\u1E00-\u1EFF]+$/, en = /\S/, ze = new re();
ze.equals = function(e, r) {
  return this.options.ignoreCase && (e = e.toLowerCase(), r = r.toLowerCase()), e === r || this.options.ignoreWhitespace && !en.test(e) && !en.test(r);
};
ze.tokenize = function(e) {
  for (var r = e.split(/([^\S\r\n]+|[()[\]{}'"\r\n]|\b)/), o = 0; o < r.length - 1; o++)
    !r[o + 1] && r[o + 2] && qe.test(r[o]) && qe.test(r[o + 2]) && (r[o] += r[o + 2], r.splice(o + 1, 2), o--);
  return r;
};
function Wn(e, r, o) {
  return o = Kn(o, {
    ignoreWhitespace: !0
  }), ze.diff(e, r, o);
}
var gn = new re();
gn.tokenize = function(e) {
  var r = [], o = e.split(/(\n|\r\n)/);
  o[o.length - 1] || o.pop();
  for (var u = 0; u < o.length; u++) {
    var h = o[u];
    u % 2 && !this.options.newlineIsToken ? r[r.length - 1] += h : (this.options.ignoreWhitespace && (h = h.trim()), r.push(h));
  }
  return r;
};
var Qn = new re();
Qn.tokenize = function(e) {
  return e.split(/(\S.+?[.!?])(?=\s+|$)/);
};
var Zn = new re();
Zn.tokenize = function(e) {
  return e.split(/([{}:;,]|\s+)/);
};
function Re(e) {
  "@babel/helpers - typeof";
  return typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? Re = function(r) {
    return typeof r;
  } : Re = function(r) {
    return r && typeof Symbol == "function" && r.constructor === Symbol && r !== Symbol.prototype ? "symbol" : typeof r;
  }, Re(e);
}
var Yn = Object.prototype.toString, _e = new re();
_e.useLongestToken = !0;
_e.tokenize = gn.tokenize;
_e.castInput = function(e) {
  var r = this.options, o = r.undefinedReplacement, u = r.stringifyReplacer, h = u === void 0 ? function(n, t) {
    return typeof t == "undefined" ? o : t;
  } : u;
  return typeof e == "string" ? e : JSON.stringify(xe(e, null, null, h), h, "  ");
};
_e.equals = function(e, r) {
  return re.prototype.equals.call(_e, e.replace(/,([\r\n])/g, "$1"), r.replace(/,([\r\n])/g, "$1"));
};
function xe(e, r, o, u, h) {
  r = r || [], o = o || [], u && (e = u(h, e));
  var n;
  for (n = 0; n < r.length; n += 1)
    if (r[n] === e)
      return o[n];
  var t;
  if (Yn.call(e) === "[object Array]") {
    for (r.push(e), t = new Array(e.length), o.push(t), n = 0; n < e.length; n += 1)
      t[n] = xe(e[n], r, o, u, h);
    return r.pop(), o.pop(), t;
  }
  if (e && e.toJSON && (e = e.toJSON()), Re(e) === "object" && e !== null) {
    r.push(e), t = {}, o.push(t);
    var i = [], a;
    for (a in e)
      e.hasOwnProperty(a) && i.push(a);
    for (i.sort(), n = 0; n < i.length; n += 1)
      a = i[n], t[a] = xe(e[a], r, o, u, a);
    r.pop(), o.pop();
  } else
    t = e;
  return t;
}
var He = new re();
He.tokenize = function(e) {
  return e.slice();
};
He.join = He.removeEmpty = function(e) {
  return e;
};
function Xn(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var hn = { exports: {} };
(function(e) {
  var r = function() {
    this.Diff_Timeout = 1, this.Diff_EditCost = 4, this.Match_Threshold = 0.5, this.Match_Distance = 1e3, this.Patch_DeleteThreshold = 0.5, this.Patch_Margin = 4, this.Match_MaxBits = 32;
  }, o = -1, u = 1, h = 0;
  r.Diff = function(n, t) {
    return [n, t];
  }, r.prototype.diff_main = function(n, t, i, a) {
    typeof a == "undefined" && (this.Diff_Timeout <= 0 ? a = Number.MAX_VALUE : a = (/* @__PURE__ */ new Date()).getTime() + this.Diff_Timeout * 1e3);
    var s = a;
    if (n == null || t == null)
      throw new Error("Null input. (diff_main)");
    if (n == t)
      return n ? [new r.Diff(h, n)] : [];
    typeof i == "undefined" && (i = !0);
    var l = i, c = this.diff_commonPrefix(n, t), f = n.substring(0, c);
    n = n.substring(c), t = t.substring(c), c = this.diff_commonSuffix(n, t);
    var g = n.substring(n.length - c);
    n = n.substring(0, n.length - c), t = t.substring(0, t.length - c);
    var d = this.diff_compute_(n, t, l, s);
    return f && d.unshift(new r.Diff(h, f)), g && d.push(new r.Diff(h, g)), this.diff_cleanupMerge(d), d;
  }, r.prototype.diff_compute_ = function(n, t, i, a) {
    var s;
    if (!n)
      return [new r.Diff(u, t)];
    if (!t)
      return [new r.Diff(o, n)];
    var l = n.length > t.length ? n : t, c = n.length > t.length ? t : n, f = l.indexOf(c);
    if (f != -1)
      return s = [
        new r.Diff(u, l.substring(0, f)),
        new r.Diff(h, c),
        new r.Diff(
          u,
          l.substring(f + c.length)
        )
      ], n.length > t.length && (s[0][0] = s[2][0] = o), s;
    if (c.length == 1)
      return [
        new r.Diff(o, n),
        new r.Diff(u, t)
      ];
    var g = this.diff_halfMatch_(n, t);
    if (g) {
      var d = g[0], p = g[1], b = g[2], E = g[3], v = g[4], w = this.diff_main(d, b, i, a), m = this.diff_main(p, E, i, a);
      return w.concat(
        [new r.Diff(h, v)],
        m
      );
    }
    return i && n.length > 100 && t.length > 100 ? this.diff_lineMode_(n, t, a) : this.diff_bisect_(n, t, a);
  }, r.prototype.diff_lineMode_ = function(n, t, i) {
    var a = this.diff_linesToChars_(n, t);
    n = a.chars1, t = a.chars2;
    var s = a.lineArray, l = this.diff_main(n, t, !1, i);
    this.diff_charsToLines_(l, s), this.diff_cleanupSemantic(l), l.push(new r.Diff(h, ""));
    for (var c = 0, f = 0, g = 0, d = "", p = ""; c < l.length; ) {
      switch (l[c][0]) {
        case u:
          g++, p += l[c][1];
          break;
        case o:
          f++, d += l[c][1];
          break;
        case h:
          if (f >= 1 && g >= 1) {
            l.splice(
              c - f - g,
              f + g
            ), c = c - f - g;
            for (var b = this.diff_main(d, p, !1, i), E = b.length - 1; E >= 0; E--)
              l.splice(c, 0, b[E]);
            c = c + b.length;
          }
          g = 0, f = 0, d = "", p = "";
          break;
      }
      c++;
    }
    return l.pop(), l;
  }, r.prototype.diff_bisect_ = function(n, t, i) {
    for (var a = n.length, s = t.length, l = Math.ceil((a + s) / 2), c = l, f = 2 * l, g = new Array(f), d = new Array(f), p = 0; p < f; p++)
      g[p] = -1, d[p] = -1;
    g[c + 1] = 0, d[c + 1] = 0;
    for (var b = a - s, E = b % 2 != 0, v = 0, w = 0, m = 0, D = 0, S = 0; S < l && !((/* @__PURE__ */ new Date()).getTime() > i); S++) {
      for (var A = -S + v; A <= S - w; A += 2) {
        var I = c + A, R;
        A == -S || A != S && g[I - 1] < g[I + 1] ? R = g[I + 1] : R = g[I - 1] + 1;
        for (var F = R - A; R < a && F < s && n.charAt(R) == t.charAt(F); )
          R++, F++;
        if (g[I] = R, R > a)
          w += 2;
        else if (F > s)
          v += 2;
        else if (E) {
          var U = c + b - A;
          if (U >= 0 && U < f && d[U] != -1) {
            var $ = a - d[U];
            if (R >= $)
              return this.diff_bisectSplit_(n, t, R, F, i);
          }
        }
      }
      for (var G = -S + m; G <= S - D; G += 2) {
        var U = c + G, $;
        G == -S || G != S && d[U - 1] < d[U + 1] ? $ = d[U + 1] : $ = d[U - 1] + 1;
        for (var Q = $ - G; $ < a && Q < s && n.charAt(a - $ - 1) == t.charAt(s - Q - 1); )
          $++, Q++;
        if (d[U] = $, $ > a)
          D += 2;
        else if (Q > s)
          m += 2;
        else if (!E) {
          var I = c + b - G;
          if (I >= 0 && I < f && g[I] != -1) {
            var R = g[I], F = c + R - I;
            if ($ = a - $, R >= $)
              return this.diff_bisectSplit_(n, t, R, F, i);
          }
        }
      }
    }
    return [
      new r.Diff(o, n),
      new r.Diff(u, t)
    ];
  }, r.prototype.diff_bisectSplit_ = function(n, t, i, a, s) {
    var l = n.substring(0, i), c = t.substring(0, a), f = n.substring(i), g = t.substring(a), d = this.diff_main(l, c, !1, s), p = this.diff_main(f, g, !1, s);
    return d.concat(p);
  }, r.prototype.diff_linesToChars_ = function(n, t) {
    var i = [], a = {};
    i[0] = "";
    function s(g) {
      for (var d = "", p = 0, b = -1, E = i.length; b < g.length - 1; ) {
        b = g.indexOf(`
`, p), b == -1 && (b = g.length - 1);
        var v = g.substring(p, b + 1);
        (a.hasOwnProperty ? a.hasOwnProperty(v) : a[v] !== void 0) ? d += String.fromCharCode(a[v]) : (E == l && (v = g.substring(p), b = g.length), d += String.fromCharCode(E), a[v] = E, i[E++] = v), p = b + 1;
      }
      return d;
    }
    var l = 4e4, c = s(n);
    l = 65535;
    var f = s(t);
    return { chars1: c, chars2: f, lineArray: i };
  }, r.prototype.diff_charsToLines_ = function(n, t) {
    for (var i = 0; i < n.length; i++) {
      for (var a = n[i][1], s = [], l = 0; l < a.length; l++)
        s[l] = t[a.charCodeAt(l)];
      n[i][1] = s.join("");
    }
  }, r.prototype.diff_commonPrefix = function(n, t) {
    if (!n || !t || n.charAt(0) != t.charAt(0))
      return 0;
    for (var i = 0, a = Math.min(n.length, t.length), s = a, l = 0; i < s; )
      n.substring(l, s) == t.substring(l, s) ? (i = s, l = i) : a = s, s = Math.floor((a - i) / 2 + i);
    return s;
  }, r.prototype.diff_commonSuffix = function(n, t) {
    if (!n || !t || n.charAt(n.length - 1) != t.charAt(t.length - 1))
      return 0;
    for (var i = 0, a = Math.min(n.length, t.length), s = a, l = 0; i < s; )
      n.substring(n.length - s, n.length - l) == t.substring(t.length - s, t.length - l) ? (i = s, l = i) : a = s, s = Math.floor((a - i) / 2 + i);
    return s;
  }, r.prototype.diff_commonOverlap_ = function(n, t) {
    var i = n.length, a = t.length;
    if (i == 0 || a == 0)
      return 0;
    i > a ? n = n.substring(i - a) : i < a && (t = t.substring(0, i));
    var s = Math.min(i, a);
    if (n == t)
      return s;
    for (var l = 0, c = 1; ; ) {
      var f = n.substring(s - c), g = t.indexOf(f);
      if (g == -1)
        return l;
      c += g, (g == 0 || n.substring(s - c) == t.substring(0, c)) && (l = c, c++);
    }
  }, r.prototype.diff_halfMatch_ = function(n, t) {
    if (this.Diff_Timeout <= 0)
      return null;
    var i = n.length > t.length ? n : t, a = n.length > t.length ? t : n;
    if (i.length < 4 || a.length * 2 < i.length)
      return null;
    var s = this;
    function l(w, m, D) {
      for (var S = w.substring(D, D + Math.floor(w.length / 4)), A = -1, I = "", R, F, U, $; (A = m.indexOf(S, A + 1)) != -1; ) {
        var G = s.diff_commonPrefix(
          w.substring(D),
          m.substring(A)
        ), Q = s.diff_commonSuffix(
          w.substring(0, D),
          m.substring(0, A)
        );
        I.length < Q + G && (I = m.substring(A - Q, A) + m.substring(A, A + G), R = w.substring(0, D - Q), F = w.substring(D + G), U = m.substring(0, A - Q), $ = m.substring(A + G));
      }
      return I.length * 2 >= w.length ? [
        R,
        F,
        U,
        $,
        I
      ] : null;
    }
    var c = l(
      i,
      a,
      Math.ceil(i.length / 4)
    ), f = l(
      i,
      a,
      Math.ceil(i.length / 2)
    ), g;
    if (!c && !f)
      return null;
    f ? c ? g = c[4].length > f[4].length ? c : f : g = f : g = c;
    var d, p, b, E;
    n.length > t.length ? (d = g[0], p = g[1], b = g[2], E = g[3]) : (b = g[0], E = g[1], d = g[2], p = g[3]);
    var v = g[4];
    return [d, p, b, E, v];
  }, r.prototype.diff_cleanupSemantic = function(n) {
    for (var t = !1, i = [], a = 0, s = null, l = 0, c = 0, f = 0, g = 0, d = 0; l < n.length; )
      n[l][0] == h ? (i[a++] = l, c = g, f = d, g = 0, d = 0, s = n[l][1]) : (n[l][0] == u ? g += n[l][1].length : d += n[l][1].length, s && s.length <= Math.max(c, f) && s.length <= Math.max(
        g,
        d
      ) && (n.splice(
        i[a - 1],
        0,
        new r.Diff(o, s)
      ), n[i[a - 1] + 1][0] = u, a--, a--, l = a > 0 ? i[a - 1] : -1, c = 0, f = 0, g = 0, d = 0, s = null, t = !0)), l++;
    for (t && this.diff_cleanupMerge(n), this.diff_cleanupSemanticLossless(n), l = 1; l < n.length; ) {
      if (n[l - 1][0] == o && n[l][0] == u) {
        var p = n[l - 1][1], b = n[l][1], E = this.diff_commonOverlap_(p, b), v = this.diff_commonOverlap_(b, p);
        E >= v ? (E >= p.length / 2 || E >= b.length / 2) && (n.splice(l, 0, new r.Diff(
          h,
          b.substring(0, E)
        )), n[l - 1][1] = p.substring(0, p.length - E), n[l + 1][1] = b.substring(E), l++) : (v >= p.length / 2 || v >= b.length / 2) && (n.splice(l, 0, new r.Diff(
          h,
          p.substring(0, v)
        )), n[l - 1][0] = u, n[l - 1][1] = b.substring(0, b.length - v), n[l + 1][0] = o, n[l + 1][1] = p.substring(v), l++), l++;
      }
      l++;
    }
  }, r.prototype.diff_cleanupSemanticLossless = function(n) {
    function t(v, w) {
      if (!v || !w)
        return 6;
      var m = v.charAt(v.length - 1), D = w.charAt(0), S = m.match(r.nonAlphaNumericRegex_), A = D.match(r.nonAlphaNumericRegex_), I = S && m.match(r.whitespaceRegex_), R = A && D.match(r.whitespaceRegex_), F = I && m.match(r.linebreakRegex_), U = R && D.match(r.linebreakRegex_), $ = F && v.match(r.blanklineEndRegex_), G = U && w.match(r.blanklineStartRegex_);
      return $ || G ? 5 : F || U ? 4 : S && !I && R ? 3 : I || R ? 2 : S || A ? 1 : 0;
    }
    for (var i = 1; i < n.length - 1; ) {
      if (n[i - 1][0] == h && n[i + 1][0] == h) {
        var a = n[i - 1][1], s = n[i][1], l = n[i + 1][1], c = this.diff_commonSuffix(a, s);
        if (c) {
          var f = s.substring(s.length - c);
          a = a.substring(0, a.length - c), s = f + s.substring(0, s.length - c), l = f + l;
        }
        for (var g = a, d = s, p = l, b = t(a, s) + t(s, l); s.charAt(0) === l.charAt(0); ) {
          a += s.charAt(0), s = s.substring(1) + l.charAt(0), l = l.substring(1);
          var E = t(a, s) + t(s, l);
          E >= b && (b = E, g = a, d = s, p = l);
        }
        n[i - 1][1] != g && (g ? n[i - 1][1] = g : (n.splice(i - 1, 1), i--), n[i][1] = d, p ? n[i + 1][1] = p : (n.splice(i + 1, 1), i--));
      }
      i++;
    }
  }, r.nonAlphaNumericRegex_ = /[^a-zA-Z0-9]/, r.whitespaceRegex_ = /\s/, r.linebreakRegex_ = /[\r\n]/, r.blanklineEndRegex_ = /\n\r?\n$/, r.blanklineStartRegex_ = /^\r?\n\r?\n/, r.prototype.diff_cleanupEfficiency = function(n) {
    for (var t = !1, i = [], a = 0, s = null, l = 0, c = !1, f = !1, g = !1, d = !1; l < n.length; )
      n[l][0] == h ? (n[l][1].length < this.Diff_EditCost && (g || d) ? (i[a++] = l, c = g, f = d, s = n[l][1]) : (a = 0, s = null), g = d = !1) : (n[l][0] == o ? d = !0 : g = !0, s && (c && f && g && d || s.length < this.Diff_EditCost / 2 && c + f + g + d == 3) && (n.splice(
        i[a - 1],
        0,
        new r.Diff(o, s)
      ), n[i[a - 1] + 1][0] = u, a--, s = null, c && f ? (g = d = !0, a = 0) : (a--, l = a > 0 ? i[a - 1] : -1, g = d = !1), t = !0)), l++;
    t && this.diff_cleanupMerge(n);
  }, r.prototype.diff_cleanupMerge = function(n) {
    n.push(new r.Diff(h, ""));
    for (var t = 0, i = 0, a = 0, s = "", l = "", c; t < n.length; )
      switch (n[t][0]) {
        case u:
          a++, l += n[t][1], t++;
          break;
        case o:
          i++, s += n[t][1], t++;
          break;
        case h:
          i + a > 1 ? (i !== 0 && a !== 0 && (c = this.diff_commonPrefix(l, s), c !== 0 && (t - i - a > 0 && n[t - i - a - 1][0] == h ? n[t - i - a - 1][1] += l.substring(0, c) : (n.splice(0, 0, new r.Diff(
            h,
            l.substring(0, c)
          )), t++), l = l.substring(c), s = s.substring(c)), c = this.diff_commonSuffix(l, s), c !== 0 && (n[t][1] = l.substring(l.length - c) + n[t][1], l = l.substring(0, l.length - c), s = s.substring(0, s.length - c))), t -= i + a, n.splice(t, i + a), s.length && (n.splice(
            t,
            0,
            new r.Diff(o, s)
          ), t++), l.length && (n.splice(
            t,
            0,
            new r.Diff(u, l)
          ), t++), t++) : t !== 0 && n[t - 1][0] == h ? (n[t - 1][1] += n[t][1], n.splice(t, 1)) : t++, a = 0, i = 0, s = "", l = "";
          break;
      }
    n[n.length - 1][1] === "" && n.pop();
    var f = !1;
    for (t = 1; t < n.length - 1; )
      n[t - 1][0] == h && n[t + 1][0] == h && (n[t][1].substring(n[t][1].length - n[t - 1][1].length) == n[t - 1][1] ? (n[t][1] = n[t - 1][1] + n[t][1].substring(0, n[t][1].length - n[t - 1][1].length), n[t + 1][1] = n[t - 1][1] + n[t + 1][1], n.splice(t - 1, 1), f = !0) : n[t][1].substring(0, n[t + 1][1].length) == n[t + 1][1] && (n[t - 1][1] += n[t + 1][1], n[t][1] = n[t][1].substring(n[t + 1][1].length) + n[t + 1][1], n.splice(t + 1, 1), f = !0)), t++;
    f && this.diff_cleanupMerge(n);
  }, r.prototype.diff_xIndex = function(n, t) {
    var i = 0, a = 0, s = 0, l = 0, c;
    for (c = 0; c < n.length && (n[c][0] !== u && (i += n[c][1].length), n[c][0] !== o && (a += n[c][1].length), !(i > t)); c++)
      s = i, l = a;
    return n.length != c && n[c][0] === o ? l : l + (t - s);
  }, r.prototype.diff_prettyHtml = function(n) {
    for (var t = [], i = /&/g, a = /</g, s = />/g, l = /\n/g, c = 0; c < n.length; c++) {
      var f = n[c][0], g = n[c][1], d = g.replace(i, "&amp;").replace(a, "&lt;").replace(s, "&gt;").replace(l, "&para;<br>");
      switch (f) {
        case u:
          t[c] = '<ins style="background:#e6ffe6;">' + d + "</ins>";
          break;
        case o:
          t[c] = '<del style="background:#ffe6e6;">' + d + "</del>";
          break;
        case h:
          t[c] = "<span>" + d + "</span>";
          break;
      }
    }
    return t.join("");
  }, r.prototype.diff_text1 = function(n) {
    for (var t = [], i = 0; i < n.length; i++)
      n[i][0] !== u && (t[i] = n[i][1]);
    return t.join("");
  }, r.prototype.diff_text2 = function(n) {
    for (var t = [], i = 0; i < n.length; i++)
      n[i][0] !== o && (t[i] = n[i][1]);
    return t.join("");
  }, r.prototype.diff_levenshtein = function(n) {
    for (var t = 0, i = 0, a = 0, s = 0; s < n.length; s++) {
      var l = n[s][0], c = n[s][1];
      switch (l) {
        case u:
          i += c.length;
          break;
        case o:
          a += c.length;
          break;
        case h:
          t += Math.max(i, a), i = 0, a = 0;
          break;
      }
    }
    return t += Math.max(i, a), t;
  }, r.prototype.diff_toDelta = function(n) {
    for (var t = [], i = 0; i < n.length; i++)
      switch (n[i][0]) {
        case u:
          t[i] = "+" + encodeURI(n[i][1]);
          break;
        case o:
          t[i] = "-" + n[i][1].length;
          break;
        case h:
          t[i] = "=" + n[i][1].length;
          break;
      }
    return t.join("	").replace(/%20/g, " ");
  }, r.prototype.diff_fromDelta = function(n, t) {
    for (var i = [], a = 0, s = 0, l = t.split(/\t/g), c = 0; c < l.length; c++) {
      var f = l[c].substring(1);
      switch (l[c].charAt(0)) {
        case "+":
          try {
            i[a++] = new r.Diff(u, decodeURI(f));
          } catch (p) {
            throw new Error("Illegal escape in diff_fromDelta: " + f);
          }
          break;
        case "-":
        case "=":
          var g = parseInt(f, 10);
          if (isNaN(g) || g < 0)
            throw new Error("Invalid number in diff_fromDelta: " + f);
          var d = n.substring(s, s += g);
          l[c].charAt(0) == "=" ? i[a++] = new r.Diff(h, d) : i[a++] = new r.Diff(o, d);
          break;
        default:
          if (l[c])
            throw new Error("Invalid diff operation in diff_fromDelta: " + l[c]);
      }
    }
    if (s != n.length)
      throw new Error("Delta length (" + s + ") does not equal source text length (" + n.length + ").");
    return i;
  }, r.prototype.match_main = function(n, t, i) {
    if (n == null || t == null || i == null)
      throw new Error("Null input. (match_main)");
    return i = Math.max(0, Math.min(i, n.length)), n == t ? 0 : n.length ? n.substring(i, i + t.length) == t ? i : this.match_bitap_(n, t, i) : -1;
  }, r.prototype.match_bitap_ = function(n, t, i) {
    if (t.length > this.Match_MaxBits)
      throw new Error("Pattern too long for this browser.");
    var a = this.match_alphabet_(t), s = this;
    function l(R, F) {
      var U = R / t.length, $ = Math.abs(i - F);
      return s.Match_Distance ? U + $ / s.Match_Distance : $ ? 1 : U;
    }
    var c = this.Match_Threshold, f = n.indexOf(t, i);
    f != -1 && (c = Math.min(l(0, f), c), f = n.lastIndexOf(t, i + t.length), f != -1 && (c = Math.min(l(0, f), c)));
    var g = 1 << t.length - 1;
    f = -1;
    for (var d, p, b = t.length + n.length, E, v = 0; v < t.length; v++) {
      for (d = 0, p = b; d < p; )
        l(v, i + p) <= c ? d = p : b = p, p = Math.floor((b - d) / 2 + d);
      b = p;
      var w = Math.max(1, i - p + 1), m = Math.min(i + p, n.length) + t.length, D = Array(m + 2);
      D[m + 1] = (1 << v) - 1;
      for (var S = m; S >= w; S--) {
        var A = a[n.charAt(S - 1)];
        if (v === 0 ? D[S] = (D[S + 1] << 1 | 1) & A : D[S] = (D[S + 1] << 1 | 1) & A | ((E[S + 1] | E[S]) << 1 | 1) | E[S + 1], D[S] & g) {
          var I = l(v, S - 1);
          if (I <= c)
            if (c = I, f = S - 1, f > i)
              w = Math.max(1, 2 * i - f);
            else
              break;
        }
      }
      if (l(v + 1, i) > c)
        break;
      E = D;
    }
    return f;
  }, r.prototype.match_alphabet_ = function(n) {
    for (var t = {}, i = 0; i < n.length; i++)
      t[n.charAt(i)] = 0;
    for (var i = 0; i < n.length; i++)
      t[n.charAt(i)] |= 1 << n.length - i - 1;
    return t;
  }, r.prototype.patch_addContext_ = function(n, t) {
    if (t.length != 0) {
      if (n.start2 === null)
        throw Error("patch not initialized");
      for (var i = t.substring(n.start2, n.start2 + n.length1), a = 0; t.indexOf(i) != t.lastIndexOf(i) && i.length < this.Match_MaxBits - this.Patch_Margin - this.Patch_Margin; )
        a += this.Patch_Margin, i = t.substring(
          n.start2 - a,
          n.start2 + n.length1 + a
        );
      a += this.Patch_Margin;
      var s = t.substring(n.start2 - a, n.start2);
      s && n.diffs.unshift(new r.Diff(h, s));
      var l = t.substring(
        n.start2 + n.length1,
        n.start2 + n.length1 + a
      );
      l && n.diffs.push(new r.Diff(h, l)), n.start1 -= s.length, n.start2 -= s.length, n.length1 += s.length + l.length, n.length2 += s.length + l.length;
    }
  }, r.prototype.patch_make = function(n, t, i) {
    var a, s;
    if (typeof n == "string" && typeof t == "string" && typeof i == "undefined")
      a = /** @type {string} */
      n, s = this.diff_main(
        a,
        /** @type {string} */
        t,
        !0
      ), s.length > 2 && (this.diff_cleanupSemantic(s), this.diff_cleanupEfficiency(s));
    else if (n && typeof n == "object" && typeof t == "undefined" && typeof i == "undefined")
      s = /** @type {!Array.<!diff_match_patch.Diff>} */
      n, a = this.diff_text1(s);
    else if (typeof n == "string" && t && typeof t == "object" && typeof i == "undefined")
      a = /** @type {string} */
      n, s = /** @type {!Array.<!diff_match_patch.Diff>} */
      t;
    else if (typeof n == "string" && typeof t == "string" && i && typeof i == "object")
      a = /** @type {string} */
      n, s = /** @type {!Array.<!diff_match_patch.Diff>} */
      i;
    else
      throw new Error("Unknown call format to patch_make.");
    if (s.length === 0)
      return [];
    for (var l = [], c = new r.patch_obj(), f = 0, g = 0, d = 0, p = a, b = a, E = 0; E < s.length; E++) {
      var v = s[E][0], w = s[E][1];
      switch (!f && v !== h && (c.start1 = g, c.start2 = d), v) {
        case u:
          c.diffs[f++] = s[E], c.length2 += w.length, b = b.substring(0, d) + w + b.substring(d);
          break;
        case o:
          c.length1 += w.length, c.diffs[f++] = s[E], b = b.substring(0, d) + b.substring(d + w.length);
          break;
        case h:
          w.length <= 2 * this.Patch_Margin && f && s.length != E + 1 ? (c.diffs[f++] = s[E], c.length1 += w.length, c.length2 += w.length) : w.length >= 2 * this.Patch_Margin && f && (this.patch_addContext_(c, p), l.push(c), c = new r.patch_obj(), f = 0, p = b, g = d);
          break;
      }
      v !== u && (g += w.length), v !== o && (d += w.length);
    }
    return f && (this.patch_addContext_(c, p), l.push(c)), l;
  }, r.prototype.patch_deepCopy = function(n) {
    for (var t = [], i = 0; i < n.length; i++) {
      var a = n[i], s = new r.patch_obj();
      s.diffs = [];
      for (var l = 0; l < a.diffs.length; l++)
        s.diffs[l] = new r.Diff(a.diffs[l][0], a.diffs[l][1]);
      s.start1 = a.start1, s.start2 = a.start2, s.length1 = a.length1, s.length2 = a.length2, t[i] = s;
    }
    return t;
  }, r.prototype.patch_apply = function(n, t) {
    if (n.length == 0)
      return [t, []];
    n = this.patch_deepCopy(n);
    var i = this.patch_addPadding(n);
    t = i + t + i, this.patch_splitMax(n);
    for (var a = 0, s = [], l = 0; l < n.length; l++) {
      var c = n[l].start2 + a, f = this.diff_text1(n[l].diffs), g, d = -1;
      if (f.length > this.Match_MaxBits ? (g = this.match_main(
        t,
        f.substring(0, this.Match_MaxBits),
        c
      ), g != -1 && (d = this.match_main(
        t,
        f.substring(f.length - this.Match_MaxBits),
        c + f.length - this.Match_MaxBits
      ), (d == -1 || g >= d) && (g = -1))) : g = this.match_main(t, f, c), g == -1)
        s[l] = !1, a -= n[l].length2 - n[l].length1;
      else {
        s[l] = !0, a = g - c;
        var p;
        if (d == -1 ? p = t.substring(g, g + f.length) : p = t.substring(g, d + this.Match_MaxBits), f == p)
          t = t.substring(0, g) + this.diff_text2(n[l].diffs) + t.substring(g + f.length);
        else {
          var b = this.diff_main(f, p, !1);
          if (f.length > this.Match_MaxBits && this.diff_levenshtein(b) / f.length > this.Patch_DeleteThreshold)
            s[l] = !1;
          else {
            this.diff_cleanupSemanticLossless(b);
            for (var E = 0, v, w = 0; w < n[l].diffs.length; w++) {
              var m = n[l].diffs[w];
              m[0] !== h && (v = this.diff_xIndex(b, E)), m[0] === u ? t = t.substring(0, g + v) + m[1] + t.substring(g + v) : m[0] === o && (t = t.substring(0, g + v) + t.substring(g + this.diff_xIndex(
                b,
                E + m[1].length
              ))), m[0] !== o && (E += m[1].length);
            }
          }
        }
      }
    }
    return t = t.substring(i.length, t.length - i.length), [t, s];
  }, r.prototype.patch_addPadding = function(n) {
    for (var t = this.Patch_Margin, i = "", a = 1; a <= t; a++)
      i += String.fromCharCode(a);
    for (var a = 0; a < n.length; a++)
      n[a].start1 += t, n[a].start2 += t;
    var s = n[0], l = s.diffs;
    if (l.length == 0 || l[0][0] != h)
      l.unshift(new r.Diff(h, i)), s.start1 -= t, s.start2 -= t, s.length1 += t, s.length2 += t;
    else if (t > l[0][1].length) {
      var c = t - l[0][1].length;
      l[0][1] = i.substring(l[0][1].length) + l[0][1], s.start1 -= c, s.start2 -= c, s.length1 += c, s.length2 += c;
    }
    if (s = n[n.length - 1], l = s.diffs, l.length == 0 || l[l.length - 1][0] != h)
      l.push(new r.Diff(h, i)), s.length1 += t, s.length2 += t;
    else if (t > l[l.length - 1][1].length) {
      var c = t - l[l.length - 1][1].length;
      l[l.length - 1][1] += i.substring(0, c), s.length1 += c, s.length2 += c;
    }
    return i;
  }, r.prototype.patch_splitMax = function(n) {
    for (var t = this.Match_MaxBits, i = 0; i < n.length; i++)
      if (!(n[i].length1 <= t)) {
        var a = n[i];
        n.splice(i--, 1);
        for (var s = a.start1, l = a.start2, c = ""; a.diffs.length !== 0; ) {
          var f = new r.patch_obj(), g = !0;
          for (f.start1 = s - c.length, f.start2 = l - c.length, c !== "" && (f.length1 = f.length2 = c.length, f.diffs.push(new r.Diff(h, c))); a.diffs.length !== 0 && f.length1 < t - this.Patch_Margin; ) {
            var d = a.diffs[0][0], p = a.diffs[0][1];
            d === u ? (f.length2 += p.length, l += p.length, f.diffs.push(a.diffs.shift()), g = !1) : d === o && f.diffs.length == 1 && f.diffs[0][0] == h && p.length > 2 * t ? (f.length1 += p.length, s += p.length, g = !1, f.diffs.push(new r.Diff(d, p)), a.diffs.shift()) : (p = p.substring(
              0,
              t - f.length1 - this.Patch_Margin
            ), f.length1 += p.length, s += p.length, d === h ? (f.length2 += p.length, l += p.length) : g = !1, f.diffs.push(new r.Diff(d, p)), p == a.diffs[0][1] ? a.diffs.shift() : a.diffs[0][1] = a.diffs[0][1].substring(p.length));
          }
          c = this.diff_text2(f.diffs), c = c.substring(c.length - this.Patch_Margin);
          var b = this.diff_text1(a.diffs).substring(0, this.Patch_Margin);
          b !== "" && (f.length1 += b.length, f.length2 += b.length, f.diffs.length !== 0 && f.diffs[f.diffs.length - 1][0] === h ? f.diffs[f.diffs.length - 1][1] += b : f.diffs.push(new r.Diff(h, b))), g || n.splice(++i, 0, f);
        }
      }
  }, r.prototype.patch_toText = function(n) {
    for (var t = [], i = 0; i < n.length; i++)
      t[i] = n[i];
    return t.join("");
  }, r.prototype.patch_fromText = function(n) {
    var t = [];
    if (!n)
      return t;
    for (var i = n.split(`
`), a = 0, s = /^@@ -(\d+),?(\d*) \+(\d+),?(\d*) @@$/; a < i.length; ) {
      var l = i[a].match(s);
      if (!l)
        throw new Error("Invalid patch string: " + i[a]);
      var c = new r.patch_obj();
      for (t.push(c), c.start1 = parseInt(l[1], 10), l[2] === "" ? (c.start1--, c.length1 = 1) : l[2] == "0" ? c.length1 = 0 : (c.start1--, c.length1 = parseInt(l[2], 10)), c.start2 = parseInt(l[3], 10), l[4] === "" ? (c.start2--, c.length2 = 1) : l[4] == "0" ? c.length2 = 0 : (c.start2--, c.length2 = parseInt(l[4], 10)), a++; a < i.length; ) {
        var f = i[a].charAt(0);
        try {
          var g = decodeURI(i[a].substring(1));
        } catch (d) {
          throw new Error("Illegal escape in patch_fromText: " + g);
        }
        if (f == "-")
          c.diffs.push(new r.Diff(o, g));
        else if (f == "+")
          c.diffs.push(new r.Diff(u, g));
        else if (f == " ")
          c.diffs.push(new r.Diff(h, g));
        else {
          if (f == "@")
            break;
          if (f !== "")
            throw new Error('Invalid patch mode "' + f + '" in: ' + g);
        }
        a++;
      }
    }
    return t;
  }, r.patch_obj = function() {
    this.diffs = [], this.start1 = null, this.start2 = null, this.length1 = 0, this.length2 = 0;
  }, r.patch_obj.prototype.toString = function() {
    var n, t;
    this.length1 === 0 ? n = this.start1 + ",0" : this.length1 == 1 ? n = this.start1 + 1 : n = this.start1 + 1 + "," + this.length1, this.length2 === 0 ? t = this.start2 + ",0" : this.length2 == 1 ? t = this.start2 + 1 : t = this.start2 + 1 + "," + this.length2;
    for (var i = ["@@ -" + n + " +" + t + ` @@
`], a, s = 0; s < this.diffs.length; s++) {
      switch (this.diffs[s][0]) {
        case u:
          a = "+";
          break;
        case o:
          a = "-";
          break;
        case h:
          a = " ";
          break;
      }
      i[s + 1] = a + encodeURI(this.diffs[s][1]) + `
`;
    }
    return i.join("").replace(/%20/g, " ");
  }, e.exports = r, e.exports.diff_match_patch = r, e.exports.DIFF_DELETE = o, e.exports.DIFF_INSERT = u, e.exports.DIFF_EQUAL = h;
})(hn);
var Ue = hn.exports, Ge = { exports: {} };
function Ke(e) {
  return e instanceof Map ? e.clear = e.delete = e.set = function() {
    throw new Error("map is read-only");
  } : e instanceof Set && (e.add = e.clear = e.delete = function() {
    throw new Error("set is read-only");
  }), Object.freeze(e), Object.getOwnPropertyNames(e).forEach(function(r) {
    var o = e[r];
    typeof o == "object" && !Object.isFrozen(o) && Ke(o);
  }), e;
}
Ge.exports = Ke;
Ge.exports.default = Ke;
class nn {
  /**
   * @param {CompiledMode} mode
   */
  constructor(r) {
    r.data === void 0 && (r.data = {}), this.data = r.data, this.isMatchIgnored = !1;
  }
  ignoreMatch() {
    this.isMatchIgnored = !0;
  }
}
function dn(e) {
  return e.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&#x27;");
}
function ae(e, ...r) {
  const o = /* @__PURE__ */ Object.create(null);
  for (const u in e)
    o[u] = e[u];
  return r.forEach(function(u) {
    for (const h in u)
      o[h] = u[h];
  }), /** @type {T} */
  o;
}
const jn = "</span>", tn = (e) => !!e.scope || e.sublanguage && e.language, Vn = (e, { prefix: r }) => {
  if (e.includes(".")) {
    const o = e.split(".");
    return [
      `${r}${o.shift()}`,
      ...o.map((u, h) => `${u}${"_".repeat(h + 1)}`)
    ].join(" ");
  }
  return `${r}${e}`;
};
class Jn {
  /**
   * Creates a new HTMLRenderer
   *
   * @param {Tree} parseTree - the parse tree (must support `walk` API)
   * @param {{classPrefix: string}} options
   */
  constructor(r, o) {
    this.buffer = "", this.classPrefix = o.classPrefix, r.walk(this);
  }
  /**
   * Adds texts to the output stream
   *
   * @param {string} text */
  addText(r) {
    this.buffer += dn(r);
  }
  /**
   * Adds a node open to the output stream (if needed)
   *
   * @param {Node} node */
  openNode(r) {
    if (!tn(r))
      return;
    let o = "";
    r.sublanguage ? o = `language-${r.language}` : o = Vn(r.scope, { prefix: this.classPrefix }), this.span(o);
  }
  /**
   * Adds a node close to the output stream (if needed)
   *
   * @param {Node} node */
  closeNode(r) {
    tn(r) && (this.buffer += jn);
  }
  /**
   * returns the accumulated buffer
  */
  value() {
    return this.buffer;
  }
  // helpers
  /**
   * Builds a span element
   *
   * @param {string} className */
  span(r) {
    this.buffer += `<span class="${r}">`;
  }
}
const rn = (e = {}) => {
  const r = { children: [] };
  return Object.assign(r, e), r;
};
class We {
  constructor() {
    this.rootNode = rn(), this.stack = [this.rootNode];
  }
  get top() {
    return this.stack[this.stack.length - 1];
  }
  get root() {
    return this.rootNode;
  }
  /** @param {Node} node */
  add(r) {
    this.top.children.push(r);
  }
  /** @param {string} scope */
  openNode(r) {
    const o = rn({ scope: r });
    this.add(o), this.stack.push(o);
  }
  closeNode() {
    if (this.stack.length > 1)
      return this.stack.pop();
  }
  closeAllNodes() {
    for (; this.closeNode(); )
      ;
  }
  toJSON() {
    return JSON.stringify(this.rootNode, null, 4);
  }
  /**
   * @typedef { import("./html_renderer").Renderer } Renderer
   * @param {Renderer} builder
   */
  walk(r) {
    return this.constructor._walk(r, this.rootNode);
  }
  /**
   * @param {Renderer} builder
   * @param {Node} node
   */
  static _walk(r, o) {
    return typeof o == "string" ? r.addText(o) : o.children && (r.openNode(o), o.children.forEach((u) => this._walk(r, u)), r.closeNode(o)), r;
  }
  /**
   * @param {Node} node
   */
  static _collapse(r) {
    typeof r != "string" && r.children && (r.children.every((o) => typeof o == "string") ? r.children = [r.children.join("")] : r.children.forEach((o) => {
      We._collapse(o);
    }));
  }
}
class qn extends We {
  /**
   * @param {*} options
   */
  constructor(r) {
    super(), this.options = r;
  }
  /**
   * @param {string} text
   * @param {string} scope
   */
  addKeyword(r, o) {
    r !== "" && (this.openNode(o), this.addText(r), this.closeNode());
  }
  /**
   * @param {string} text
   */
  addText(r) {
    r !== "" && this.add(r);
  }
  /**
   * @param {Emitter & {root: DataNode}} emitter
   * @param {string} name
   */
  addSublanguage(r, o) {
    const u = r.root;
    u.sublanguage = !0, u.language = o, this.add(u);
  }
  toHTML() {
    return new Jn(this, this.options).value();
  }
  finalize() {
    return !0;
  }
}
function Ee(e) {
  return e ? typeof e == "string" ? e : e.source : null;
}
function pn(e) {
  return ce("(?=", e, ")");
}
function et(e) {
  return ce("(?:", e, ")*");
}
function nt(e) {
  return ce("(?:", e, ")?");
}
function ce(...e) {
  return e.map((o) => Ee(o)).join("");
}
function tt(e) {
  const r = e[e.length - 1];
  return typeof r == "object" && r.constructor === Object ? (e.splice(e.length - 1, 1), r) : {};
}
function Qe(...e) {
  return "(" + (tt(e).capture ? "" : "?:") + e.map((u) => Ee(u)).join("|") + ")";
}
function bn(e) {
  return new RegExp(e.toString() + "|").exec("").length - 1;
}
function rt(e, r) {
  const o = e && e.exec(r);
  return o && o.index === 0;
}
const it = /\[(?:[^\\\]]|\\.)*\]|\(\??|\\([1-9][0-9]*)|\\./;
function Ze(e, { joinWith: r }) {
  let o = 0;
  return e.map((u) => {
    o += 1;
    const h = o;
    let n = Ee(u), t = "";
    for (; n.length > 0; ) {
      const i = it.exec(n);
      if (!i) {
        t += n;
        break;
      }
      t += n.substring(0, i.index), n = n.substring(i.index + i[0].length), i[0][0] === "\\" && i[1] ? t += "\\" + String(Number(i[1]) + h) : (t += i[0], i[0] === "(" && o++);
    }
    return t;
  }).map((u) => `(${u})`).join(r);
}
const st = /\b\B/, _n = "[a-zA-Z]\\w*", Ye = "[a-zA-Z_]\\w*", En = "\\b\\d+(\\.\\d+)?", vn = "(-?)(\\b0[xX][a-fA-F0-9]+|(\\b\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)", mn = "\\b(0b[01]+)", at = "!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|-|-=|/=|/|:|;|<<|<<=|<=|<|===|==|=|>>>=|>>=|>=|>>>|>>|>|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~", ot = (e = {}) => {
  const r = /^#![ ]*\//;
  return e.binary && (e.begin = ce(
    r,
    /.*\b/,
    e.binary,
    /\b.*/
  )), ae({
    scope: "meta",
    begin: r,
    end: /$/,
    relevance: 0,
    /** @type {ModeCallback} */
    "on:begin": (o, u) => {
      o.index !== 0 && u.ignoreMatch();
    }
  }, e);
}, ve = {
  begin: "\\\\[\\s\\S]",
  relevance: 0
}, lt = {
  scope: "string",
  begin: "'",
  end: "'",
  illegal: "\\n",
  contains: [ve]
}, ct = {
  scope: "string",
  begin: '"',
  end: '"',
  illegal: "\\n",
  contains: [ve]
}, ut = {
  begin: /\b(a|an|the|are|I'm|isn't|don't|doesn't|won't|but|just|should|pretty|simply|enough|gonna|going|wtf|so|such|will|you|your|they|like|more)\b/
}, Ie = function(e, r, o = {}) {
  const u = ae(
    {
      scope: "comment",
      begin: e,
      end: r,
      contains: []
    },
    o
  );
  u.contains.push({
    scope: "doctag",
    // hack to avoid the space from being included. the space is necessary to
    // match here to prevent the plain text rule below from gobbling up doctags
    begin: "[ ]*(?=(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):)",
    end: /(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):/,
    excludeBegin: !0,
    relevance: 0
  });
  const h = Qe(
    // list of common 1 and 2 letter words in English
    "I",
    "a",
    "is",
    "so",
    "us",
    "to",
    "at",
    "if",
    "in",
    "it",
    "on",
    // note: this is not an exhaustive list of contractions, just popular ones
    /[A-Za-z]+['](d|ve|re|ll|t|s|n)/,
    // contractions - can't we'd they're let's, etc
    /[A-Za-z]+[-][a-z]+/,
    // `no-way`, etc.
    /[A-Za-z][a-z]{2,}/
    // allow capitalized words at beginning of sentences
  );
  return u.contains.push(
    {
      // TODO: how to include ", (, ) without breaking grammars that use these for
      // comment delimiters?
      // begin: /[ ]+([()"]?([A-Za-z'-]{3,}|is|a|I|so|us|[tT][oO]|at|if|in|it|on)[.]?[()":]?([.][ ]|[ ]|\))){3}/
      // ---
      // this tries to find sequences of 3 english words in a row (without any
      // "programming" type syntax) this gives us a strong signal that we've
      // TRULY found a comment - vs perhaps scanning with the wrong language.
      // It's possible to find something that LOOKS like the start of the
      // comment - but then if there is no readable text - good chance it is a
      // false match and not a comment.
      //
      // for a visual example please see:
      // https://github.com/highlightjs/highlight.js/issues/2827
      begin: ce(
        /[ ]+/,
        // necessary to prevent us gobbling up doctags like /* @author Bob Mcgill */
        "(",
        h,
        /[.]?[:]?([.][ ]|[ ])/,
        "){3}"
      )
      // look for 3 words in a row
    }
  ), u;
}, ft = Ie("//", "$"), gt = Ie("/\\*", "\\*/"), ht = Ie("#", "$"), dt = {
  scope: "number",
  begin: En,
  relevance: 0
}, pt = {
  scope: "number",
  begin: vn,
  relevance: 0
}, bt = {
  scope: "number",
  begin: mn,
  relevance: 0
}, _t = {
  // this outer rule makes sure we actually have a WHOLE regex and not simply
  // an expression such as:
  //
  //     3 / something
  //
  // (which will then blow up when regex's `illegal` sees the newline)
  begin: /(?=\/[^/\n]*\/)/,
  contains: [{
    scope: "regexp",
    begin: /\//,
    end: /\/[gimuy]*/,
    illegal: /\n/,
    contains: [
      ve,
      {
        begin: /\[/,
        end: /\]/,
        relevance: 0,
        contains: [ve]
      }
    ]
  }]
}, Et = {
  scope: "title",
  begin: _n,
  relevance: 0
}, vt = {
  scope: "title",
  begin: Ye,
  relevance: 0
}, mt = {
  // excludes method names from keyword processing
  begin: "\\.\\s*" + Ye,
  relevance: 0
}, yt = function(e) {
  return Object.assign(
    e,
    {
      /** @type {ModeCallback} */
      "on:begin": (r, o) => {
        o.data._beginMatch = r[1];
      },
      /** @type {ModeCallback} */
      "on:end": (r, o) => {
        o.data._beginMatch !== r[1] && o.ignoreMatch();
      }
    }
  );
};
var Te = /* @__PURE__ */ Object.freeze({
  __proto__: null,
  MATCH_NOTHING_RE: st,
  IDENT_RE: _n,
  UNDERSCORE_IDENT_RE: Ye,
  NUMBER_RE: En,
  C_NUMBER_RE: vn,
  BINARY_NUMBER_RE: mn,
  RE_STARTERS_RE: at,
  SHEBANG: ot,
  BACKSLASH_ESCAPE: ve,
  APOS_STRING_MODE: lt,
  QUOTE_STRING_MODE: ct,
  PHRASAL_WORDS_MODE: ut,
  COMMENT: Ie,
  C_LINE_COMMENT_MODE: ft,
  C_BLOCK_COMMENT_MODE: gt,
  HASH_COMMENT_MODE: ht,
  NUMBER_MODE: dt,
  C_NUMBER_MODE: pt,
  BINARY_NUMBER_MODE: bt,
  REGEXP_MODE: _t,
  TITLE_MODE: Et,
  UNDERSCORE_TITLE_MODE: vt,
  METHOD_GUARD: mt,
  END_SAME_AS_BEGIN: yt
});
function wt(e, r) {
  e.input[e.index - 1] === "." && r.ignoreMatch();
}
function Nt(e, r) {
  e.className !== void 0 && (e.scope = e.className, delete e.className);
}
function Mt(e, r) {
  r && e.beginKeywords && (e.begin = "\\b(" + e.beginKeywords.split(" ").join("|") + ")(?!\\.)(?=\\b|\\s)", e.__beforeBegin = wt, e.keywords = e.keywords || e.beginKeywords, delete e.beginKeywords, e.relevance === void 0 && (e.relevance = 0));
}
function At(e, r) {
  Array.isArray(e.illegal) && (e.illegal = Qe(...e.illegal));
}
function Tt(e, r) {
  if (e.match) {
    if (e.begin || e.end)
      throw new Error("begin & end are not supported with match");
    e.begin = e.match, delete e.match;
  }
}
function Dt(e, r) {
  e.relevance === void 0 && (e.relevance = 1);
}
const St = (e, r) => {
  if (!e.beforeMatch)
    return;
  if (e.starts)
    throw new Error("beforeMatch cannot be used with starts");
  const o = Object.assign({}, e);
  Object.keys(e).forEach((u) => {
    delete e[u];
  }), e.keywords = o.keywords, e.begin = ce(o.beforeMatch, pn(o.begin)), e.starts = {
    relevance: 0,
    contains: [
      Object.assign(o, { endsParent: !0 })
    ]
  }, e.relevance = 0, delete o.beforeMatch;
}, Rt = [
  "of",
  "and",
  "for",
  "in",
  "not",
  "or",
  "if",
  "then",
  "parent",
  // common variable name
  "list",
  // common variable name
  "value"
  // common variable name
], Ot = "keyword";
function yn(e, r, o = Ot) {
  const u = /* @__PURE__ */ Object.create(null);
  return typeof e == "string" ? h(o, e.split(" ")) : Array.isArray(e) ? h(o, e) : Object.keys(e).forEach(function(n) {
    Object.assign(
      u,
      yn(e[n], r, n)
    );
  }), u;
  function h(n, t) {
    r && (t = t.map((i) => i.toLowerCase())), t.forEach(function(i) {
      const a = i.split("|");
      u[a[0]] = [n, Lt(a[0], a[1])];
    });
  }
}
function Lt(e, r) {
  return r ? Number(r) : Ct(e) ? 0 : 1;
}
function Ct(e) {
  return Rt.includes(e.toLowerCase());
}
const sn = {}, le = (e) => {
  console.error(e);
}, an = (e, ...r) => {
  console.log(`WARN: ${e}`, ...r);
}, ue = (e, r) => {
  sn[`${e}/${r}`] || (console.log(`Deprecated as of ${e}. ${r}`), sn[`${e}/${r}`] = !0);
}, Le = new Error();
function wn(e, r, { key: o }) {
  let u = 0;
  const h = e[o], n = {}, t = {};
  for (let i = 1; i <= r.length; i++)
    t[i + u] = h[i], n[i + u] = !0, u += bn(r[i - 1]);
  e[o] = t, e[o]._emit = n, e[o]._multi = !0;
}
function It(e) {
  if (Array.isArray(e.begin)) {
    if (e.skip || e.excludeBegin || e.returnBegin)
      throw le("skip, excludeBegin, returnBegin not compatible with beginScope: {}"), Le;
    if (typeof e.beginScope != "object" || e.beginScope === null)
      throw le("beginScope must be object"), Le;
    wn(e, e.begin, { key: "beginScope" }), e.begin = Ze(e.begin, { joinWith: "" });
  }
}
function kt(e) {
  if (Array.isArray(e.end)) {
    if (e.skip || e.excludeEnd || e.returnEnd)
      throw le("skip, excludeEnd, returnEnd not compatible with endScope: {}"), Le;
    if (typeof e.endScope != "object" || e.endScope === null)
      throw le("endScope must be object"), Le;
    wn(e, e.end, { key: "endScope" }), e.end = Ze(e.end, { joinWith: "" });
  }
}
function Bt(e) {
  e.scope && typeof e.scope == "object" && e.scope !== null && (e.beginScope = e.scope, delete e.scope);
}
function $t(e) {
  Bt(e), typeof e.beginScope == "string" && (e.beginScope = { _wrap: e.beginScope }), typeof e.endScope == "string" && (e.endScope = { _wrap: e.endScope }), It(e), kt(e);
}
function Ut(e) {
  function r(t, i) {
    return new RegExp(
      Ee(t),
      "m" + (e.case_insensitive ? "i" : "") + (e.unicodeRegex ? "u" : "") + (i ? "g" : "")
    );
  }
  class o {
    constructor() {
      this.matchIndexes = {}, this.regexes = [], this.matchAt = 1, this.position = 0;
    }
    // @ts-ignore
    addRule(i, a) {
      a.position = this.position++, this.matchIndexes[this.matchAt] = a, this.regexes.push([a, i]), this.matchAt += bn(i) + 1;
    }
    compile() {
      this.regexes.length === 0 && (this.exec = () => null);
      const i = this.regexes.map((a) => a[1]);
      this.matcherRe = r(Ze(i, { joinWith: "|" }), !0), this.lastIndex = 0;
    }
    /** @param {string} s */
    exec(i) {
      this.matcherRe.lastIndex = this.lastIndex;
      const a = this.matcherRe.exec(i);
      if (!a)
        return null;
      const s = a.findIndex((c, f) => f > 0 && c !== void 0), l = this.matchIndexes[s];
      return a.splice(0, s), Object.assign(a, l);
    }
  }
  class u {
    constructor() {
      this.rules = [], this.multiRegexes = [], this.count = 0, this.lastIndex = 0, this.regexIndex = 0;
    }
    // @ts-ignore
    getMatcher(i) {
      if (this.multiRegexes[i])
        return this.multiRegexes[i];
      const a = new o();
      return this.rules.slice(i).forEach(([s, l]) => a.addRule(s, l)), a.compile(), this.multiRegexes[i] = a, a;
    }
    resumingScanAtSamePosition() {
      return this.regexIndex !== 0;
    }
    considerAll() {
      this.regexIndex = 0;
    }
    // @ts-ignore
    addRule(i, a) {
      this.rules.push([i, a]), a.type === "begin" && this.count++;
    }
    /** @param {string} s */
    exec(i) {
      const a = this.getMatcher(this.regexIndex);
      a.lastIndex = this.lastIndex;
      let s = a.exec(i);
      if (this.resumingScanAtSamePosition() && !(s && s.index === this.lastIndex)) {
        const l = this.getMatcher(0);
        l.lastIndex = this.lastIndex + 1, s = l.exec(i);
      }
      return s && (this.regexIndex += s.position + 1, this.regexIndex === this.count && this.considerAll()), s;
    }
  }
  function h(t) {
    const i = new u();
    return t.contains.forEach((a) => i.addRule(a.begin, { rule: a, type: "begin" })), t.terminatorEnd && i.addRule(t.terminatorEnd, { type: "end" }), t.illegal && i.addRule(t.illegal, { type: "illegal" }), i;
  }
  function n(t, i) {
    const a = (
      /** @type CompiledMode */
      t
    );
    if (t.isCompiled)
      return a;
    [
      Nt,
      // do this early so compiler extensions generally don't have to worry about
      // the distinction between match/begin
      Tt,
      $t,
      St
    ].forEach((l) => l(t, i)), e.compilerExtensions.forEach((l) => l(t, i)), t.__beforeBegin = null, [
      Mt,
      // do this later so compiler extensions that come earlier have access to the
      // raw array if they wanted to perhaps manipulate it, etc.
      At,
      // default to 1 relevance if not specified
      Dt
    ].forEach((l) => l(t, i)), t.isCompiled = !0;
    let s = null;
    return typeof t.keywords == "object" && t.keywords.$pattern && (t.keywords = Object.assign({}, t.keywords), s = t.keywords.$pattern, delete t.keywords.$pattern), s = s || /\w+/, t.keywords && (t.keywords = yn(t.keywords, e.case_insensitive)), a.keywordPatternRe = r(s, !0), i && (t.begin || (t.begin = /\B|\b/), a.beginRe = r(a.begin), !t.end && !t.endsWithParent && (t.end = /\B|\b/), t.end && (a.endRe = r(a.end)), a.terminatorEnd = Ee(a.end) || "", t.endsWithParent && i.terminatorEnd && (a.terminatorEnd += (t.end ? "|" : "") + i.terminatorEnd)), t.illegal && (a.illegalRe = r(
      /** @type {RegExp | string} */
      t.illegal
    )), t.contains || (t.contains = []), t.contains = [].concat(...t.contains.map(function(l) {
      return Pt(l === "self" ? t : l);
    })), t.contains.forEach(function(l) {
      n(
        /** @type Mode */
        l,
        a
      );
    }), t.starts && n(t.starts, i), a.matcher = h(a), a;
  }
  if (e.compilerExtensions || (e.compilerExtensions = []), e.contains && e.contains.includes("self"))
    throw new Error("ERR: contains `self` is not supported at the top-level of a language.  See documentation.");
  return e.classNameAliases = ae(e.classNameAliases || {}), n(
    /** @type Mode */
    e
  );
}
function Nn(e) {
  return e ? e.endsWithParent || Nn(e.starts) : !1;
}
function Pt(e) {
  return e.variants && !e.cachedVariants && (e.cachedVariants = e.variants.map(function(r) {
    return ae(e, { variants: null }, r);
  })), e.cachedVariants ? e.cachedVariants : Nn(e) ? ae(e, { starts: e.starts ? ae(e.starts) : null }) : Object.isFrozen(e) ? ae(e) : e;
}
var xt = "11.7.0";
class Ht extends Error {
  constructor(r, o) {
    super(r), this.name = "HTMLInjectionError", this.html = o;
  }
}
const Pe = dn, on = ae, ln = Symbol("nomatch"), Ft = 7, zt = function(e) {
  const r = /* @__PURE__ */ Object.create(null), o = /* @__PURE__ */ Object.create(null), u = [];
  let h = !0;
  const n = "Could not find the language '{}', did you forget to load/include a language module?", t = { disableAutodetect: !0, name: "Plain text", contains: [] };
  let i = {
    ignoreUnescapedHTML: !1,
    throwUnescapedHTML: !1,
    noHighlightRe: /^(no-?highlight)$/i,
    languageDetectRe: /\blang(?:uage)?-([\w-]+)\b/i,
    classPrefix: "hljs-",
    cssSelector: "pre code",
    languages: null,
    // beta configuration options, subject to change, welcome to discuss
    // https://github.com/highlightjs/highlight.js/issues/1086
    __emitter: qn
  };
  function a(_) {
    return i.noHighlightRe.test(_);
  }
  function s(_) {
    let M = _.className + " ";
    M += _.parentNode ? _.parentNode.className : "";
    const L = i.languageDetectRe.exec(M);
    if (L) {
      const k = R(L[1]);
      return k || (an(n.replace("{}", L[1])), an("Falling back to no-highlight mode for this block.", _)), k ? L[1] : "no-highlight";
    }
    return M.split(/\s+/).find((k) => a(k) || R(k));
  }
  function l(_, M, L) {
    let k = "", H = "";
    typeof M == "object" ? (k = _, L = M.ignoreIllegals, H = M.language) : (ue("10.7.0", "highlight(lang, code, ...args) has been deprecated."), ue("10.7.0", `Please use highlight(code, options) instead.
https://github.com/highlightjs/highlight.js/issues/2277`), H = _, k = M), L === void 0 && (L = !0);
    const Y = {
      code: k,
      language: H
    };
    Q("before:highlight", Y);
    const q = Y.result ? Y.result : c(Y.language, Y.code, L);
    return q.code = Y.code, Q("after:highlight", q), q;
  }
  function c(_, M, L, k) {
    const H = /* @__PURE__ */ Object.create(null);
    function Y(y, T) {
      return y.keywords[T];
    }
    function q() {
      if (!O.keywords) {
        K.addText(x);
        return;
      }
      let y = 0;
      O.keywordPatternRe.lastIndex = 0;
      let T = O.keywordPatternRe.exec(x), C = "";
      for (; T; ) {
        C += x.substring(y, T.index);
        const B = se.case_insensitive ? T[0].toLowerCase() : T[0], Z = Y(O, B);
        if (Z) {
          const [ee, Bn] = Z;
          if (K.addText(C), C = "", H[B] = (H[B] || 0) + 1, H[B] <= Ft && (Ae += Bn), ee.startsWith("_"))
            C += T[0];
          else {
            const $n = se.classNameAliases[ee] || ee;
            K.addKeyword(T[0], $n);
          }
        } else
          C += T[0];
        y = O.keywordPatternRe.lastIndex, T = O.keywordPatternRe.exec(x);
      }
      C += x.substring(y), K.addText(C);
    }
    function Ne() {
      if (x === "")
        return;
      let y = null;
      if (typeof O.subLanguage == "string") {
        if (!r[O.subLanguage]) {
          K.addText(x);
          return;
        }
        y = c(O.subLanguage, x, !0, Je[O.subLanguage]), Je[O.subLanguage] = /** @type {CompiledMode} */
        y._top;
      } else
        y = g(x, O.subLanguage.length ? O.subLanguage : null);
      O.relevance > 0 && (Ae += y.relevance), K.addSublanguage(y._emitter, y.language);
    }
    function X() {
      O.subLanguage != null ? Ne() : q(), x = "";
    }
    function ie(y, T) {
      let C = 1;
      const B = T.length - 1;
      for (; C <= B; ) {
        if (!y._emit[C]) {
          C++;
          continue;
        }
        const Z = se.classNameAliases[y[C]] || y[C], ee = T[C];
        Z ? K.addKeyword(ee, Z) : (x = ee, q(), x = ""), C++;
      }
    }
    function Xe(y, T) {
      return y.scope && typeof y.scope == "string" && K.openNode(se.classNameAliases[y.scope] || y.scope), y.beginScope && (y.beginScope._wrap ? (K.addKeyword(x, se.classNameAliases[y.beginScope._wrap] || y.beginScope._wrap), x = "") : y.beginScope._multi && (ie(y.beginScope, T), x = "")), O = Object.create(y, { parent: { value: O } }), O;
    }
    function je(y, T, C) {
      let B = rt(y.endRe, C);
      if (B) {
        if (y["on:end"]) {
          const Z = new nn(y);
          y["on:end"](T, Z), Z.isMatchIgnored && (B = !1);
        }
        if (B) {
          for (; y.endsParent && y.parent; )
            y = y.parent;
          return y;
        }
      }
      if (y.endsWithParent)
        return je(y.parent, T, C);
    }
    function On(y) {
      return O.matcher.regexIndex === 0 ? (x += y[0], 1) : ($e = !0, 0);
    }
    function Ln(y) {
      const T = y[0], C = y.rule, B = new nn(C), Z = [C.__beforeBegin, C["on:begin"]];
      for (const ee of Z)
        if (ee && (ee(y, B), B.isMatchIgnored))
          return On(T);
      return C.skip ? x += T : (C.excludeBegin && (x += T), X(), !C.returnBegin && !C.excludeBegin && (x = T)), Xe(C, y), C.returnBegin ? 0 : T.length;
    }
    function Cn(y) {
      const T = y[0], C = M.substring(y.index), B = je(O, y, C);
      if (!B)
        return ln;
      const Z = O;
      O.endScope && O.endScope._wrap ? (X(), K.addKeyword(T, O.endScope._wrap)) : O.endScope && O.endScope._multi ? (X(), ie(O.endScope, y)) : Z.skip ? x += T : (Z.returnEnd || Z.excludeEnd || (x += T), X(), Z.excludeEnd && (x = T));
      do
        O.scope && K.closeNode(), !O.skip && !O.subLanguage && (Ae += O.relevance), O = O.parent;
      while (O !== B.parent);
      return B.starts && Xe(B.starts, y), Z.returnEnd ? 0 : T.length;
    }
    function In() {
      const y = [];
      for (let T = O; T !== se; T = T.parent)
        T.scope && y.unshift(T.scope);
      y.forEach((T) => K.openNode(T));
    }
    let Me = {};
    function Ve(y, T) {
      const C = T && T[0];
      if (x += y, C == null)
        return X(), 0;
      if (Me.type === "begin" && T.type === "end" && Me.index === T.index && C === "") {
        if (x += M.slice(T.index, T.index + 1), !h) {
          const B = new Error(`0 width match regex (${_})`);
          throw B.languageName = _, B.badRule = Me.rule, B;
        }
        return 1;
      }
      if (Me = T, T.type === "begin")
        return Ln(T);
      if (T.type === "illegal" && !L) {
        const B = new Error('Illegal lexeme "' + C + '" for mode "' + (O.scope || "<unnamed>") + '"');
        throw B.mode = O, B;
      } else if (T.type === "end") {
        const B = Cn(T);
        if (B !== ln)
          return B;
      }
      if (T.type === "illegal" && C === "")
        return 1;
      if (Be > 1e5 && Be > T.index * 3)
        throw new Error("potential infinite loop, way more iterations than matches");
      return x += C, C.length;
    }
    const se = R(_);
    if (!se)
      throw le(n.replace("{}", _)), new Error('Unknown language: "' + _ + '"');
    const kn = Ut(se);
    let ke = "", O = k || kn;
    const Je = {}, K = new i.__emitter(i);
    In();
    let x = "", Ae = 0, oe = 0, Be = 0, $e = !1;
    try {
      for (O.matcher.considerAll(); ; ) {
        Be++, $e ? $e = !1 : O.matcher.considerAll(), O.matcher.lastIndex = oe;
        const y = O.matcher.exec(M);
        if (!y)
          break;
        const T = M.substring(oe, y.index), C = Ve(T, y);
        oe = y.index + C;
      }
      return Ve(M.substring(oe)), K.closeAllNodes(), K.finalize(), ke = K.toHTML(), {
        language: _,
        value: ke,
        relevance: Ae,
        illegal: !1,
        _emitter: K,
        _top: O
      };
    } catch (y) {
      if (y.message && y.message.includes("Illegal"))
        return {
          language: _,
          value: Pe(M),
          illegal: !0,
          relevance: 0,
          _illegalBy: {
            message: y.message,
            index: oe,
            context: M.slice(oe - 100, oe + 100),
            mode: y.mode,
            resultSoFar: ke
          },
          _emitter: K
        };
      if (h)
        return {
          language: _,
          value: Pe(M),
          illegal: !1,
          relevance: 0,
          errorRaised: y,
          _emitter: K,
          _top: O
        };
      throw y;
    }
  }
  function f(_) {
    const M = {
      value: Pe(_),
      illegal: !1,
      relevance: 0,
      _top: t,
      _emitter: new i.__emitter(i)
    };
    return M._emitter.addText(_), M;
  }
  function g(_, M) {
    M = M || i.languages || Object.keys(r);
    const L = f(_), k = M.filter(R).filter(U).map(
      (X) => c(X, _, !1)
    );
    k.unshift(L);
    const H = k.sort((X, ie) => {
      if (X.relevance !== ie.relevance)
        return ie.relevance - X.relevance;
      if (X.language && ie.language) {
        if (R(X.language).supersetOf === ie.language)
          return 1;
        if (R(ie.language).supersetOf === X.language)
          return -1;
      }
      return 0;
    }), [Y, q] = H, Ne = Y;
    return Ne.secondBest = q, Ne;
  }
  function d(_, M, L) {
    const k = M && o[M] || L;
    _.classList.add("hljs"), _.classList.add(`language-${k}`);
  }
  function p(_) {
    let M = null;
    const L = s(_);
    if (a(L))
      return;
    if (Q(
      "before:highlightElement",
      { el: _, language: L }
    ), _.children.length > 0 && (i.ignoreUnescapedHTML || (console.warn("One of your code blocks includes unescaped HTML. This is a potentially serious security risk."), console.warn("https://github.com/highlightjs/highlight.js/wiki/security"), console.warn("The element with unescaped HTML:"), console.warn(_)), i.throwUnescapedHTML))
      throw new Ht(
        "One of your code blocks includes unescaped HTML.",
        _.innerHTML
      );
    M = _;
    const k = M.textContent, H = L ? l(k, { language: L, ignoreIllegals: !0 }) : g(k);
    _.innerHTML = H.value, d(_, L, H.language), _.result = {
      language: H.language,
      // TODO: remove with version 11.0
      re: H.relevance,
      relevance: H.relevance
    }, H.secondBest && (_.secondBest = {
      language: H.secondBest.language,
      relevance: H.secondBest.relevance
    }), Q("after:highlightElement", { el: _, result: H, text: k });
  }
  function b(_) {
    i = on(i, _);
  }
  const E = () => {
    m(), ue("10.6.0", "initHighlighting() deprecated.  Use highlightAll() now.");
  };
  function v() {
    m(), ue("10.6.0", "initHighlightingOnLoad() deprecated.  Use highlightAll() now.");
  }
  let w = !1;
  function m() {
    if (document.readyState === "loading") {
      w = !0;
      return;
    }
    document.querySelectorAll(i.cssSelector).forEach(p);
  }
  function D() {
    w && m();
  }
  typeof window != "undefined" && window.addEventListener && window.addEventListener("DOMContentLoaded", D, !1);
  function S(_, M) {
    let L = null;
    try {
      L = M(e);
    } catch (k) {
      if (le("Language definition for '{}' could not be registered.".replace("{}", _)), h)
        le(k);
      else
        throw k;
      L = t;
    }
    L.name || (L.name = _), r[_] = L, L.rawDefinition = M.bind(null, e), L.aliases && F(L.aliases, { languageName: _ });
  }
  function A(_) {
    delete r[_];
    for (const M of Object.keys(o))
      o[M] === _ && delete o[M];
  }
  function I() {
    return Object.keys(r);
  }
  function R(_) {
    return _ = (_ || "").toLowerCase(), r[_] || r[o[_]];
  }
  function F(_, { languageName: M }) {
    typeof _ == "string" && (_ = [_]), _.forEach((L) => {
      o[L.toLowerCase()] = M;
    });
  }
  function U(_) {
    const M = R(_);
    return M && !M.disableAutodetect;
  }
  function $(_) {
    _["before:highlightBlock"] && !_["before:highlightElement"] && (_["before:highlightElement"] = (M) => {
      _["before:highlightBlock"](
        Object.assign({ block: M.el }, M)
      );
    }), _["after:highlightBlock"] && !_["after:highlightElement"] && (_["after:highlightElement"] = (M) => {
      _["after:highlightBlock"](
        Object.assign({ block: M.el }, M)
      );
    });
  }
  function G(_) {
    $(_), u.push(_);
  }
  function Q(_, M) {
    const L = _;
    u.forEach(function(k) {
      k[L] && k[L](M);
    });
  }
  function we(_) {
    return ue("10.7.0", "highlightBlock will be removed entirely in v12.0"), ue("10.7.0", "Please use highlightElement now."), p(_);
  }
  Object.assign(e, {
    highlight: l,
    highlightAuto: g,
    highlightAll: m,
    highlightElement: p,
    // TODO: Remove with v12 API
    highlightBlock: we,
    configure: b,
    initHighlighting: E,
    initHighlightingOnLoad: v,
    registerLanguage: S,
    unregisterLanguage: A,
    listLanguages: I,
    getLanguage: R,
    registerAliases: F,
    autoDetection: U,
    inherit: on,
    addPlugin: G
  }), e.debugMode = function() {
    h = !1;
  }, e.safeMode = function() {
    h = !0;
  }, e.versionString = xt, e.regex = {
    concat: ce,
    lookahead: pn,
    either: Qe,
    optional: nt,
    anyNumberOfTimes: et
  };
  for (const _ in Te)
    typeof Te[_] == "object" && Ge.exports(Te[_]);
  return Object.assign(e, Te), e;
};
var me = zt({}), Gt = me;
me.HighlightJS = me;
me.default = me;
const J = /* @__PURE__ */ Xn(Gt);
function Kt(e) {
  const r = e.regex, o = r.concat(/[\p{L}_]/u, r.optional(/[\p{L}0-9_.-]*:/u), /[\p{L}0-9_.-]*/u), u = /[\p{L}0-9._:-]+/u, h = {
    className: "symbol",
    begin: /&[a-z]+;|&#[0-9]+;|&#x[a-f0-9]+;/
  }, n = {
    begin: /\s/,
    contains: [
      {
        className: "keyword",
        begin: /#?[a-z_][a-z1-9_-]+/,
        illegal: /\n/
      }
    ]
  }, t = e.inherit(n, {
    begin: /\(/,
    end: /\)/
  }), i = e.inherit(e.APOS_STRING_MODE, { className: "string" }), a = e.inherit(e.QUOTE_STRING_MODE, { className: "string" }), s = {
    endsWithParent: !0,
    illegal: /</,
    relevance: 0,
    contains: [
      {
        className: "attr",
        begin: u,
        relevance: 0
      },
      {
        begin: /=\s*/,
        relevance: 0,
        contains: [
          {
            className: "string",
            endsParent: !0,
            variants: [
              {
                begin: /"/,
                end: /"/,
                contains: [h]
              },
              {
                begin: /'/,
                end: /'/,
                contains: [h]
              },
              { begin: /[^\s"'=<>`]+/ }
            ]
          }
        ]
      }
    ]
  };
  return {
    name: "HTML, XML",
    aliases: [
      "html",
      "xhtml",
      "rss",
      "atom",
      "xjb",
      "xsd",
      "xsl",
      "plist",
      "wsf",
      "svg"
    ],
    case_insensitive: !0,
    unicodeRegex: !0,
    contains: [
      {
        className: "meta",
        begin: /<![a-z]/,
        end: />/,
        relevance: 10,
        contains: [
          n,
          a,
          i,
          t,
          {
            begin: /\[/,
            end: /\]/,
            contains: [
              {
                className: "meta",
                begin: /<![a-z]/,
                end: />/,
                contains: [
                  n,
                  t,
                  a,
                  i
                ]
              }
            ]
          }
        ]
      },
      e.COMMENT(
        /<!--/,
        /-->/,
        { relevance: 10 }
      ),
      {
        begin: /<!\[CDATA\[/,
        end: /\]\]>/,
        relevance: 10
      },
      h,
      // xml processing instructions
      {
        className: "meta",
        end: /\?>/,
        variants: [
          {
            begin: /<\?xml/,
            relevance: 10,
            contains: [
              a
            ]
          },
          {
            begin: /<\?[a-z][a-z0-9]+/
          }
        ]
      },
      {
        className: "tag",
        /*
        The lookahead pattern (?=...) ensures that 'begin' only matches
        '<style' as a single word, followed by a whitespace or an
        ending bracket.
        */
        begin: /<style(?=\s|>)/,
        end: />/,
        keywords: { name: "style" },
        contains: [s],
        starts: {
          end: /<\/style>/,
          returnEnd: !0,
          subLanguage: [
            "css",
            "xml"
          ]
        }
      },
      {
        className: "tag",
        // See the comment in the <style tag about the lookahead pattern
        begin: /<script(?=\s|>)/,
        end: />/,
        keywords: { name: "script" },
        contains: [s],
        starts: {
          end: /<\/script>/,
          returnEnd: !0,
          subLanguage: [
            "javascript",
            "handlebars",
            "xml"
          ]
        }
      },
      // we need this for now for jSX
      {
        className: "tag",
        begin: /<>|<\/>/
      },
      // open tag
      {
        className: "tag",
        begin: r.concat(
          /</,
          r.lookahead(r.concat(
            o,
            // <tag/>
            // <tag>
            // <tag ...
            r.either(/\/>/, />/, /\s/)
          ))
        ),
        end: /\/?>/,
        contains: [
          {
            className: "name",
            begin: o,
            relevance: 0,
            starts: s
          }
        ]
      },
      // close tag
      {
        className: "tag",
        begin: r.concat(
          /<\//,
          r.lookahead(r.concat(
            o,
            />/
          ))
        ),
        contains: [
          {
            className: "name",
            begin: o,
            relevance: 0
          },
          {
            begin: />/,
            relevance: 0,
            endsParent: !0
          }
        ]
      }
    ]
  };
}
const cn = "[A-Za-z$_][0-9A-Za-z$_]*", Wt = [
  "as",
  // for exports
  "in",
  "of",
  "if",
  "for",
  "while",
  "finally",
  "var",
  "new",
  "function",
  "do",
  "return",
  "void",
  "else",
  "break",
  "catch",
  "instanceof",
  "with",
  "throw",
  "case",
  "default",
  "try",
  "switch",
  "continue",
  "typeof",
  "delete",
  "let",
  "yield",
  "const",
  "class",
  // JS handles these with a special rule
  // "get",
  // "set",
  "debugger",
  "async",
  "await",
  "static",
  "import",
  "from",
  "export",
  "extends"
], Qt = [
  "true",
  "false",
  "null",
  "undefined",
  "NaN",
  "Infinity"
], Mn = [
  // Fundamental objects
  "Object",
  "Function",
  "Boolean",
  "Symbol",
  // numbers and dates
  "Math",
  "Date",
  "Number",
  "BigInt",
  // text
  "String",
  "RegExp",
  // Indexed collections
  "Array",
  "Float32Array",
  "Float64Array",
  "Int8Array",
  "Uint8Array",
  "Uint8ClampedArray",
  "Int16Array",
  "Int32Array",
  "Uint16Array",
  "Uint32Array",
  "BigInt64Array",
  "BigUint64Array",
  // Keyed collections
  "Set",
  "Map",
  "WeakSet",
  "WeakMap",
  // Structured data
  "ArrayBuffer",
  "SharedArrayBuffer",
  "Atomics",
  "DataView",
  "JSON",
  // Control abstraction objects
  "Promise",
  "Generator",
  "GeneratorFunction",
  "AsyncFunction",
  // Reflection
  "Reflect",
  "Proxy",
  // Internationalization
  "Intl",
  // WebAssembly
  "WebAssembly"
], An = [
  "Error",
  "EvalError",
  "InternalError",
  "RangeError",
  "ReferenceError",
  "SyntaxError",
  "TypeError",
  "URIError"
], Tn = [
  "setInterval",
  "setTimeout",
  "clearInterval",
  "clearTimeout",
  "require",
  "exports",
  "eval",
  "isFinite",
  "isNaN",
  "parseFloat",
  "parseInt",
  "decodeURI",
  "decodeURIComponent",
  "encodeURI",
  "encodeURIComponent",
  "escape",
  "unescape"
], Zt = [
  "arguments",
  "this",
  "super",
  "console",
  "window",
  "document",
  "localStorage",
  "module",
  "global"
  // Node.js
], Yt = [].concat(
  Tn,
  Mn,
  An
);
function Xt(e) {
  const r = e.regex, o = (M, { after: L }) => {
    const k = "</" + M[0].slice(1);
    return M.input.indexOf(k, L) !== -1;
  }, u = cn, h = {
    begin: "<>",
    end: "</>"
  }, n = /<[A-Za-z0-9\\._:-]+\s*\/>/, t = {
    begin: /<[A-Za-z0-9\\._:-]+/,
    end: /\/[A-Za-z0-9\\._:-]+>|\/>/,
    /**
     * @param {RegExpMatchArray} match
     * @param {CallbackResponse} response
     */
    isTrulyOpeningTag: (M, L) => {
      const k = M[0].length + M.index, H = M.input[k];
      if (
        // HTML should not include another raw `<` inside a tag
        // nested type?
        // `<Array<Array<number>>`, etc.
        H === "<" || // the , gives away that this is not HTML
        // `<T, A extends keyof T, V>`
        H === ","
      ) {
        L.ignoreMatch();
        return;
      }
      H === ">" && (o(M, { after: k }) || L.ignoreMatch());
      let Y;
      const q = M.input.substring(k);
      if (Y = q.match(/^\s*=/)) {
        L.ignoreMatch();
        return;
      }
      if ((Y = q.match(/^\s+extends\s+/)) && Y.index === 0) {
        L.ignoreMatch();
        return;
      }
    }
  }, i = {
    $pattern: cn,
    keyword: Wt,
    literal: Qt,
    built_in: Yt,
    "variable.language": Zt
  }, a = "[0-9](_?[0-9])*", s = `\\.(${a})`, l = "0|[1-9](_?[0-9])*|0[0-7]*[89][0-9]*", c = {
    className: "number",
    variants: [
      // DecimalLiteral
      { begin: `(\\b(${l})((${s})|\\.)?|(${s}))[eE][+-]?(${a})\\b` },
      { begin: `\\b(${l})\\b((${s})\\b|\\.)?|(${s})\\b` },
      // DecimalBigIntegerLiteral
      { begin: "\\b(0|[1-9](_?[0-9])*)n\\b" },
      // NonDecimalIntegerLiteral
      { begin: "\\b0[xX][0-9a-fA-F](_?[0-9a-fA-F])*n?\\b" },
      { begin: "\\b0[bB][0-1](_?[0-1])*n?\\b" },
      { begin: "\\b0[oO][0-7](_?[0-7])*n?\\b" },
      // LegacyOctalIntegerLiteral (does not include underscore separators)
      // https://tc39.es/ecma262/#sec-additional-syntax-numeric-literals
      { begin: "\\b0[0-7]+n?\\b" }
    ],
    relevance: 0
  }, f = {
    className: "subst",
    begin: "\\$\\{",
    end: "\\}",
    keywords: i,
    contains: []
    // defined later
  }, g = {
    begin: "html`",
    end: "",
    starts: {
      end: "`",
      returnEnd: !1,
      contains: [
        e.BACKSLASH_ESCAPE,
        f
      ],
      subLanguage: "xml"
    }
  }, d = {
    begin: "css`",
    end: "",
    starts: {
      end: "`",
      returnEnd: !1,
      contains: [
        e.BACKSLASH_ESCAPE,
        f
      ],
      subLanguage: "css"
    }
  }, p = {
    className: "string",
    begin: "`",
    end: "`",
    contains: [
      e.BACKSLASH_ESCAPE,
      f
    ]
  }, E = {
    className: "comment",
    variants: [
      e.COMMENT(
        /\/\*\*(?!\/)/,
        "\\*/",
        {
          relevance: 0,
          contains: [
            {
              begin: "(?=@[A-Za-z]+)",
              relevance: 0,
              contains: [
                {
                  className: "doctag",
                  begin: "@[A-Za-z]+"
                },
                {
                  className: "type",
                  begin: "\\{",
                  end: "\\}",
                  excludeEnd: !0,
                  excludeBegin: !0,
                  relevance: 0
                },
                {
                  className: "variable",
                  begin: u + "(?=\\s*(-)|$)",
                  endsParent: !0,
                  relevance: 0
                },
                // eat spaces (not newlines) so we can find
                // types or variables
                {
                  begin: /(?=[^\n])\s/,
                  relevance: 0
                }
              ]
            }
          ]
        }
      ),
      e.C_BLOCK_COMMENT_MODE,
      e.C_LINE_COMMENT_MODE
    ]
  }, v = [
    e.APOS_STRING_MODE,
    e.QUOTE_STRING_MODE,
    g,
    d,
    p,
    // Skip numbers when they are part of a variable name
    { match: /\$\d+/ },
    c
    // This is intentional:
    // See https://github.com/highlightjs/highlight.js/issues/3288
    // hljs.REGEXP_MODE
  ];
  f.contains = v.concat({
    // we need to pair up {} inside our subst to prevent
    // it from ending too early by matching another }
    begin: /\{/,
    end: /\}/,
    keywords: i,
    contains: [
      "self"
    ].concat(v)
  });
  const w = [].concat(E, f.contains), m = w.concat([
    // eat recursive parens in sub expressions
    {
      begin: /\(/,
      end: /\)/,
      keywords: i,
      contains: ["self"].concat(w)
    }
  ]), D = {
    className: "params",
    begin: /\(/,
    end: /\)/,
    excludeBegin: !0,
    excludeEnd: !0,
    keywords: i,
    contains: m
  }, S = {
    variants: [
      // class Car extends vehicle
      {
        match: [
          /class/,
          /\s+/,
          u,
          /\s+/,
          /extends/,
          /\s+/,
          r.concat(u, "(", r.concat(/\./, u), ")*")
        ],
        scope: {
          1: "keyword",
          3: "title.class",
          5: "keyword",
          7: "title.class.inherited"
        }
      },
      // class Car
      {
        match: [
          /class/,
          /\s+/,
          u
        ],
        scope: {
          1: "keyword",
          3: "title.class"
        }
      }
    ]
  }, A = {
    relevance: 0,
    match: r.either(
      // Hard coded exceptions
      /\bJSON/,
      // Float32Array, OutT
      /\b[A-Z][a-z]+([A-Z][a-z]*|\d)*/,
      // CSSFactory, CSSFactoryT
      /\b[A-Z]{2,}([A-Z][a-z]+|\d)+([A-Z][a-z]*)*/,
      // FPs, FPsT
      /\b[A-Z]{2,}[a-z]+([A-Z][a-z]+|\d)*([A-Z][a-z]*)*/
      // P
      // single letters are not highlighted
      // BLAH
      // this will be flagged as a UPPER_CASE_CONSTANT instead
    ),
    className: "title.class",
    keywords: {
      _: [
        // se we still get relevance credit for JS library classes
        ...Mn,
        ...An
      ]
    }
  }, I = {
    label: "use_strict",
    className: "meta",
    relevance: 10,
    begin: /^\s*['"]use (strict|asm)['"]/
  }, R = {
    variants: [
      {
        match: [
          /function/,
          /\s+/,
          u,
          /(?=\s*\()/
        ]
      },
      // anonymous function
      {
        match: [
          /function/,
          /\s*(?=\()/
        ]
      }
    ],
    className: {
      1: "keyword",
      3: "title.function"
    },
    label: "func.def",
    contains: [D],
    illegal: /%/
  }, F = {
    relevance: 0,
    match: /\b[A-Z][A-Z_0-9]+\b/,
    className: "variable.constant"
  };
  function U(M) {
    return r.concat("(?!", M.join("|"), ")");
  }
  const $ = {
    match: r.concat(
      /\b/,
      U([
        ...Tn,
        "super",
        "import"
      ]),
      u,
      r.lookahead(/\(/)
    ),
    className: "title.function",
    relevance: 0
  }, G = {
    begin: r.concat(/\./, r.lookahead(
      r.concat(u, /(?![0-9A-Za-z$_(])/)
    )),
    end: u,
    excludeBegin: !0,
    keywords: "prototype",
    className: "property",
    relevance: 0
  }, Q = {
    match: [
      /get|set/,
      /\s+/,
      u,
      /(?=\()/
    ],
    className: {
      1: "keyword",
      3: "title.function"
    },
    contains: [
      {
        // eat to avoid empty params
        begin: /\(\)/
      },
      D
    ]
  }, we = "(\\([^()]*(\\([^()]*(\\([^()]*\\)[^()]*)*\\)[^()]*)*\\)|" + e.UNDERSCORE_IDENT_RE + ")\\s*=>", _ = {
    match: [
      /const|var|let/,
      /\s+/,
      u,
      /\s*/,
      /=\s*/,
      /(async\s*)?/,
      // async is optional
      r.lookahead(we)
    ],
    keywords: "async",
    className: {
      1: "keyword",
      3: "title.function"
    },
    contains: [
      D
    ]
  };
  return {
    name: "Javascript",
    aliases: ["js", "jsx", "mjs", "cjs"],
    keywords: i,
    // this will be extended by TypeScript
    exports: { PARAMS_CONTAINS: m, CLASS_REFERENCE: A },
    illegal: /#(?![$_A-z])/,
    contains: [
      e.SHEBANG({
        label: "shebang",
        binary: "node",
        relevance: 5
      }),
      I,
      e.APOS_STRING_MODE,
      e.QUOTE_STRING_MODE,
      g,
      d,
      p,
      E,
      // Skip numbers when they are part of a variable name
      { match: /\$\d+/ },
      c,
      A,
      {
        className: "attr",
        begin: u + r.lookahead(":"),
        relevance: 0
      },
      _,
      {
        // "value" container
        begin: "(" + e.RE_STARTERS_RE + "|\\b(case|return|throw)\\b)\\s*",
        keywords: "return throw case",
        relevance: 0,
        contains: [
          E,
          e.REGEXP_MODE,
          {
            className: "function",
            // we have to count the parens to make sure we actually have the
            // correct bounding ( ) before the =>.  There could be any number of
            // sub-expressions inside also surrounded by parens.
            begin: we,
            returnBegin: !0,
            end: "\\s*=>",
            contains: [
              {
                className: "params",
                variants: [
                  {
                    begin: e.UNDERSCORE_IDENT_RE,
                    relevance: 0
                  },
                  {
                    className: null,
                    begin: /\(\s*\)/,
                    skip: !0
                  },
                  {
                    begin: /\(/,
                    end: /\)/,
                    excludeBegin: !0,
                    excludeEnd: !0,
                    keywords: i,
                    contains: m
                  }
                ]
              }
            ]
          },
          {
            // could be a comma delimited list of params to a function call
            begin: /,/,
            relevance: 0
          },
          {
            match: /\s+/,
            relevance: 0
          },
          {
            // JSX
            variants: [
              { begin: h.begin, end: h.end },
              { match: n },
              {
                begin: t.begin,
                // we carefully check the opening tag to see if it truly
                // is a tag and not a false positive
                "on:begin": t.isTrulyOpeningTag,
                end: t.end
              }
            ],
            subLanguage: "xml",
            contains: [
              {
                begin: t.begin,
                end: t.end,
                skip: !0,
                contains: ["self"]
              }
            ]
          }
        ]
      },
      R,
      {
        // prevent this from getting swallowed up by function
        // since they appear "function like"
        beginKeywords: "while if switch catch for"
      },
      {
        // we have to count the parens to make sure we actually have the correct
        // bounding ( ).  There could be any number of sub-expressions inside
        // also surrounded by parens.
        begin: "\\b(?!function)" + e.UNDERSCORE_IDENT_RE + "\\([^()]*(\\([^()]*(\\([^()]*\\)[^()]*)*\\)[^()]*)*\\)\\s*\\{",
        // end parens
        returnBegin: !0,
        label: "func.def",
        contains: [
          D,
          e.inherit(e.TITLE_MODE, { begin: u, className: "title.function" })
        ]
      },
      // catch ... so it won't trigger the property rule below
      {
        match: /\.\.\./,
        relevance: 0
      },
      G,
      // hack: prevents detection of keywords in some circumstances
      // .keyword()
      // $keyword = x
      {
        match: "\\$" + u,
        relevance: 0
      },
      {
        match: [/\bconstructor(?=\s*\()/],
        className: { 1: "title.function" },
        contains: [D]
      },
      $,
      F,
      S,
      Q,
      {
        match: /\$[(.]/
        // relevance booster for a pattern common to JS libs: `$(something)` and `$.something`
      }
    ]
  };
}
function jt(e) {
  const r = {
    className: "attr",
    begin: /"(\\.|[^\\"\r\n])*"(?=\s*:)/,
    relevance: 1.01
  }, o = {
    match: /[{}[\],:]/,
    className: "punctuation",
    relevance: 0
  }, u = [
    "true",
    "false",
    "null"
  ], h = {
    scope: "literal",
    beginKeywords: u.join(" ")
  };
  return {
    name: "JSON",
    keywords: {
      literal: u
    },
    contains: [
      r,
      o,
      e.QUOTE_STRING_MODE,
      h,
      e.C_NUMBER_MODE,
      e.C_LINE_COMMENT_MODE,
      e.C_BLOCK_COMMENT_MODE
    ],
    illegal: "\\S"
  };
}
function Vt(e) {
  const r = "true false yes no null", o = "[\\w#;/?:@&=+$,.~*'()[\\]]+", u = {
    className: "attr",
    variants: [
      { begin: "\\w[\\w :\\/.-]*:(?=[ 	]|$)" },
      {
        // double quoted keys
        begin: '"\\w[\\w :\\/.-]*":(?=[ 	]|$)'
      },
      {
        // single quoted keys
        begin: "'\\w[\\w :\\/.-]*':(?=[ 	]|$)"
      }
    ]
  }, h = {
    className: "template-variable",
    variants: [
      {
        // jinja templates Ansible
        begin: /\{\{/,
        end: /\}\}/
      },
      {
        // Ruby i18n
        begin: /%\{/,
        end: /\}/
      }
    ]
  }, n = {
    className: "string",
    relevance: 0,
    variants: [
      {
        begin: /'/,
        end: /'/
      },
      {
        begin: /"/,
        end: /"/
      },
      { begin: /\S+/ }
    ],
    contains: [
      e.BACKSLASH_ESCAPE,
      h
    ]
  }, t = e.inherit(n, { variants: [
    {
      begin: /'/,
      end: /'/
    },
    {
      begin: /"/,
      end: /"/
    },
    { begin: /[^\s,{}[\]]+/ }
  ] }), i = "[0-9]{4}(-[0-9][0-9]){0,2}", a = "([Tt \\t][0-9][0-9]?(:[0-9][0-9]){2})?", s = "(\\.[0-9]*)?", l = "([ \\t])*(Z|[-+][0-9][0-9]?(:[0-9][0-9])?)?", c = {
    className: "number",
    begin: "\\b" + i + a + s + l + "\\b"
  }, f = {
    end: ",",
    endsWithParent: !0,
    excludeEnd: !0,
    keywords: r,
    relevance: 0
  }, g = {
    begin: /\{/,
    end: /\}/,
    contains: [f],
    illegal: "\\n",
    relevance: 0
  }, d = {
    begin: "\\[",
    end: "\\]",
    contains: [f],
    illegal: "\\n",
    relevance: 0
  }, p = [
    u,
    {
      className: "meta",
      begin: "^---\\s*$",
      relevance: 10
    },
    {
      // multi line string
      // Blocks start with a | or > followed by a newline
      //
      // Indentation of subsequent lines must be the same to
      // be considered part of the block
      className: "string",
      begin: "[\\|>]([1-9]?[+-])?[ ]*\\n( +)[^ ][^\\n]*\\n(\\2[^\\n]+\\n?)*"
    },
    {
      // Ruby/Rails erb
      begin: "<%[%=-]?",
      end: "[%-]?%>",
      subLanguage: "ruby",
      excludeBegin: !0,
      excludeEnd: !0,
      relevance: 0
    },
    {
      // named tags
      className: "type",
      begin: "!\\w+!" + o
    },
    // https://yaml.org/spec/1.2/spec.html#id2784064
    {
      // verbatim tags
      className: "type",
      begin: "!<" + o + ">"
    },
    {
      // primary tags
      className: "type",
      begin: "!" + o
    },
    {
      // secondary tags
      className: "type",
      begin: "!!" + o
    },
    {
      // fragment id &ref
      className: "meta",
      begin: "&" + e.UNDERSCORE_IDENT_RE + "$"
    },
    {
      // fragment reference *ref
      className: "meta",
      begin: "\\*" + e.UNDERSCORE_IDENT_RE + "$"
    },
    {
      // array listing
      className: "bullet",
      // TODO: remove |$ hack when we have proper look-ahead support
      begin: "-(?=[ ]|$)",
      relevance: 0
    },
    e.HASH_COMMENT_MODE,
    {
      beginKeywords: r,
      keywords: { literal: r }
    },
    c,
    // numbers are any valid C-style number that
    // sit isolated from other words
    {
      className: "number",
      begin: e.C_NUMBER_RE + "\\b",
      relevance: 0
    },
    g,
    d,
    n
  ], b = [...p];
  return b.pop(), b.push(t), f.contains = b, {
    name: "YAML",
    case_insensitive: !0,
    aliases: ["yml"],
    contains: p
  };
}
function Jt(e) {
  return {
    name: "Plain text",
    aliases: [
      "text",
      "txt"
    ],
    disableAutodetect: !0
  };
}
function qt(e) {
  const r = e.regex, o = /[\p{XID_Start}_]\p{XID_Continue}*/u, u = [
    "and",
    "as",
    "assert",
    "async",
    "await",
    "break",
    "case",
    "class",
    "continue",
    "def",
    "del",
    "elif",
    "else",
    "except",
    "finally",
    "for",
    "from",
    "global",
    "if",
    "import",
    "in",
    "is",
    "lambda",
    "match",
    "nonlocal|10",
    "not",
    "or",
    "pass",
    "raise",
    "return",
    "try",
    "while",
    "with",
    "yield"
  ], i = {
    $pattern: /[A-Za-z]\w+|__\w+__/,
    keyword: u,
    built_in: [
      "__import__",
      "abs",
      "all",
      "any",
      "ascii",
      "bin",
      "bool",
      "breakpoint",
      "bytearray",
      "bytes",
      "callable",
      "chr",
      "classmethod",
      "compile",
      "complex",
      "delattr",
      "dict",
      "dir",
      "divmod",
      "enumerate",
      "eval",
      "exec",
      "filter",
      "float",
      "format",
      "frozenset",
      "getattr",
      "globals",
      "hasattr",
      "hash",
      "help",
      "hex",
      "id",
      "input",
      "int",
      "isinstance",
      "issubclass",
      "iter",
      "len",
      "list",
      "locals",
      "map",
      "max",
      "memoryview",
      "min",
      "next",
      "object",
      "oct",
      "open",
      "ord",
      "pow",
      "print",
      "property",
      "range",
      "repr",
      "reversed",
      "round",
      "set",
      "setattr",
      "slice",
      "sorted",
      "staticmethod",
      "str",
      "sum",
      "super",
      "tuple",
      "type",
      "vars",
      "zip"
    ],
    literal: [
      "__debug__",
      "Ellipsis",
      "False",
      "None",
      "NotImplemented",
      "True"
    ],
    type: [
      "Any",
      "Callable",
      "Coroutine",
      "Dict",
      "List",
      "Literal",
      "Generic",
      "Optional",
      "Sequence",
      "Set",
      "Tuple",
      "Type",
      "Union"
    ]
  }, a = {
    className: "meta",
    begin: /^(>>>|\.\.\.) /
  }, s = {
    className: "subst",
    begin: /\{/,
    end: /\}/,
    keywords: i,
    illegal: /#/
  }, l = {
    begin: /\{\{/,
    relevance: 0
  }, c = {
    className: "string",
    contains: [e.BACKSLASH_ESCAPE],
    variants: [
      {
        begin: /([uU]|[bB]|[rR]|[bB][rR]|[rR][bB])?'''/,
        end: /'''/,
        contains: [
          e.BACKSLASH_ESCAPE,
          a
        ],
        relevance: 10
      },
      {
        begin: /([uU]|[bB]|[rR]|[bB][rR]|[rR][bB])?"""/,
        end: /"""/,
        contains: [
          e.BACKSLASH_ESCAPE,
          a
        ],
        relevance: 10
      },
      {
        begin: /([fF][rR]|[rR][fF]|[fF])'''/,
        end: /'''/,
        contains: [
          e.BACKSLASH_ESCAPE,
          a,
          l,
          s
        ]
      },
      {
        begin: /([fF][rR]|[rR][fF]|[fF])"""/,
        end: /"""/,
        contains: [
          e.BACKSLASH_ESCAPE,
          a,
          l,
          s
        ]
      },
      {
        begin: /([uU]|[rR])'/,
        end: /'/,
        relevance: 10
      },
      {
        begin: /([uU]|[rR])"/,
        end: /"/,
        relevance: 10
      },
      {
        begin: /([bB]|[bB][rR]|[rR][bB])'/,
        end: /'/
      },
      {
        begin: /([bB]|[bB][rR]|[rR][bB])"/,
        end: /"/
      },
      {
        begin: /([fF][rR]|[rR][fF]|[fF])'/,
        end: /'/,
        contains: [
          e.BACKSLASH_ESCAPE,
          l,
          s
        ]
      },
      {
        begin: /([fF][rR]|[rR][fF]|[fF])"/,
        end: /"/,
        contains: [
          e.BACKSLASH_ESCAPE,
          l,
          s
        ]
      },
      e.APOS_STRING_MODE,
      e.QUOTE_STRING_MODE
    ]
  }, f = "[0-9](_?[0-9])*", g = `(\\b(${f}))?\\.(${f})|\\b(${f})\\.`, d = `\\b|${u.join("|")}`, p = {
    className: "number",
    relevance: 0,
    variants: [
      // exponentfloat, pointfloat
      // https://docs.python.org/3.9/reference/lexical_analysis.html#floating-point-literals
      // optionally imaginary
      // https://docs.python.org/3.9/reference/lexical_analysis.html#imaginary-literals
      // Note: no leading \b because floats can start with a decimal point
      // and we don't want to mishandle e.g. `fn(.5)`,
      // no trailing \b for pointfloat because it can end with a decimal point
      // and we don't want to mishandle e.g. `0..hex()`; this should be safe
      // because both MUST contain a decimal point and so cannot be confused with
      // the interior part of an identifier
      {
        begin: `(\\b(${f})|(${g}))[eE][+-]?(${f})[jJ]?(?=${d})`
      },
      {
        begin: `(${g})[jJ]?`
      },
      // decinteger, bininteger, octinteger, hexinteger
      // https://docs.python.org/3.9/reference/lexical_analysis.html#integer-literals
      // optionally "long" in Python 2
      // https://docs.python.org/2.7/reference/lexical_analysis.html#integer-and-long-integer-literals
      // decinteger is optionally imaginary
      // https://docs.python.org/3.9/reference/lexical_analysis.html#imaginary-literals
      {
        begin: `\\b([1-9](_?[0-9])*|0+(_?0)*)[lLjJ]?(?=${d})`
      },
      {
        begin: `\\b0[bB](_?[01])+[lL]?(?=${d})`
      },
      {
        begin: `\\b0[oO](_?[0-7])+[lL]?(?=${d})`
      },
      {
        begin: `\\b0[xX](_?[0-9a-fA-F])+[lL]?(?=${d})`
      },
      // imagnumber (digitpart-based)
      // https://docs.python.org/3.9/reference/lexical_analysis.html#imaginary-literals
      {
        begin: `\\b(${f})[jJ](?=${d})`
      }
    ]
  }, b = {
    className: "comment",
    begin: r.lookahead(/# type:/),
    end: /$/,
    keywords: i,
    contains: [
      {
        // prevent keywords from coloring `type`
        begin: /# type:/
      },
      // comment within a datatype comment includes no keywords
      {
        begin: /#/,
        end: /\b\B/,
        endsWithParent: !0
      }
    ]
  }, E = {
    className: "params",
    variants: [
      // Exclude params in functions without params
      {
        className: "",
        begin: /\(\s*\)/,
        skip: !0
      },
      {
        begin: /\(/,
        end: /\)/,
        excludeBegin: !0,
        excludeEnd: !0,
        keywords: i,
        contains: [
          "self",
          a,
          p,
          c,
          e.HASH_COMMENT_MODE
        ]
      }
    ]
  };
  return s.contains = [
    c,
    p,
    a
  ], {
    name: "Python",
    aliases: [
      "py",
      "gyp",
      "ipython"
    ],
    unicodeRegex: !0,
    keywords: i,
    illegal: /(<\/|->|\?)|=>/,
    contains: [
      a,
      p,
      {
        // very common convention
        begin: /\bself\b/
      },
      {
        // eat "if" prior to string so that it won't accidentally be
        // labeled as an f-string
        beginKeywords: "if",
        relevance: 0
      },
      c,
      b,
      e.HASH_COMMENT_MODE,
      {
        match: [
          /\bdef/,
          /\s+/,
          o
        ],
        scope: {
          1: "keyword",
          3: "title.function"
        },
        contains: [E]
      },
      {
        variants: [
          {
            match: [
              /\bclass/,
              /\s+/,
              o,
              /\s*/,
              /\(\s*/,
              o,
              /\s*\)/
            ]
          },
          {
            match: [
              /\bclass/,
              /\s+/,
              o
            ]
          }
        ],
        scope: {
          1: "keyword",
          3: "title.class",
          6: "title.class.inherited"
        }
      },
      {
        className: "meta",
        begin: /^[\t ]*@/,
        end: /(?=#)|$/,
        contains: [
          p,
          E,
          c
        ]
      }
    ]
  };
}
var fe = "[0-9](_*[0-9])*", De = `\\.(${fe})`, Se = "[0-9a-fA-F](_*[0-9a-fA-F])*", un = {
  className: "number",
  variants: [
    // DecimalFloatingPointLiteral
    // including ExponentPart
    { begin: `(\\b(${fe})((${De})|\\.)?|(${De}))[eE][+-]?(${fe})[fFdD]?\\b` },
    // excluding ExponentPart
    { begin: `\\b(${fe})((${De})[fFdD]?\\b|\\.([fFdD]\\b)?)` },
    { begin: `(${De})[fFdD]?\\b` },
    { begin: `\\b(${fe})[fFdD]\\b` },
    // HexadecimalFloatingPointLiteral
    { begin: `\\b0[xX]((${Se})\\.?|(${Se})?\\.(${Se}))[pP][+-]?(${fe})[fFdD]?\\b` },
    // DecimalIntegerLiteral
    { begin: "\\b(0|[1-9](_*[0-9])*)[lL]?\\b" },
    // HexIntegerLiteral
    { begin: `\\b0[xX](${Se})[lL]?\\b` },
    // OctalIntegerLiteral
    { begin: "\\b0(_*[0-7])*[lL]?\\b" },
    // BinaryIntegerLiteral
    { begin: "\\b0[bB][01](_*[01])*[lL]?\\b" }
  ],
  relevance: 0
};
function Dn(e, r, o) {
  return o === -1 ? "" : e.replace(r, (u) => Dn(e, r, o - 1));
}
function er(e) {
  const r = e.regex, o = "[-a-zA-Z_$][-a-zA-Z_$0-9]*", u = o + Dn("(?:<" + o + "~~~(?:\\s*,\\s*" + o + "~~~)*>)?", /~~~/g, 2), a = {
    keyword: [
      "synchronized",
      "abstract",
      "private",
      "var",
      "static",
      "if",
      "const ",
      "for",
      "while",
      "strictfp",
      "finally",
      "protected",
      "import",
      "native",
      "final",
      "void",
      "enum",
      "else",
      "break",
      "transient",
      "catch",
      "instanceof",
      "volatile",
      "case",
      "assert",
      "package",
      "default",
      "public",
      "try",
      "switch",
      "continue",
      "throws",
      "protected",
      "public",
      "private",
      "module",
      "requires",
      "exports",
      "do",
      "sealed",
      "yield",
      "permits"
    ],
    literal: [
      "false",
      "true",
      "null"
    ],
    type: [
      "char",
      "boolean",
      "long",
      "float",
      "int",
      "byte",
      "short",
      "double"
    ],
    built_in: [
      "super",
      "this"
    ]
  }, s = {
    className: "meta",
    begin: "@" + o,
    contains: [
      {
        begin: /\(/,
        end: /\)/,
        contains: ["self"]
        // allow nested () inside our annotation
      }
    ]
  }, l = {
    className: "params",
    begin: /\(/,
    end: /\)/,
    keywords: a,
    relevance: 0,
    contains: [e.C_BLOCK_COMMENT_MODE],
    endsParent: !0
  };
  return {
    name: "Java",
    aliases: ["jsp"],
    keywords: a,
    illegal: /<\/|#/,
    contains: [
      e.COMMENT(
        "/\\*\\*",
        "\\*/",
        {
          relevance: 0,
          contains: [
            {
              // eat up @'s in emails to prevent them to be recognized as doctags
              begin: /\w+@/,
              relevance: 0
            },
            {
              className: "doctag",
              begin: "@[A-Za-z]+"
            }
          ]
        }
      ),
      // relevance boost
      {
        begin: /import java\.[a-z]+\./,
        keywords: "import",
        relevance: 2
      },
      e.C_LINE_COMMENT_MODE,
      e.C_BLOCK_COMMENT_MODE,
      {
        begin: /"""/,
        end: /"""/,
        className: "string",
        contains: [e.BACKSLASH_ESCAPE]
      },
      e.APOS_STRING_MODE,
      e.QUOTE_STRING_MODE,
      {
        match: [
          /\b(?:class|interface|enum|extends|implements|new)/,
          /\s+/,
          o
        ],
        className: {
          1: "keyword",
          3: "title.class"
        }
      },
      {
        // Exceptions for hyphenated keywords
        match: /non-sealed/,
        scope: "keyword"
      },
      {
        begin: [
          r.concat(/(?!else)/, o),
          /\s+/,
          o,
          /\s+/,
          /=(?!=)/
        ],
        className: {
          1: "type",
          3: "variable",
          5: "operator"
        }
      },
      {
        begin: [
          /record/,
          /\s+/,
          o
        ],
        className: {
          1: "keyword",
          3: "title.class"
        },
        contains: [
          l,
          e.C_LINE_COMMENT_MODE,
          e.C_BLOCK_COMMENT_MODE
        ]
      },
      {
        // Expression keywords prevent 'keyword Name(...)' from being
        // recognized as a function definition
        beginKeywords: "new throw return else",
        relevance: 0
      },
      {
        begin: [
          "(?:" + u + "\\s+)",
          e.UNDERSCORE_IDENT_RE,
          /\s*(?=\()/
        ],
        className: { 2: "title.function" },
        keywords: a,
        contains: [
          {
            className: "params",
            begin: /\(/,
            end: /\)/,
            keywords: a,
            relevance: 0,
            contains: [
              s,
              e.APOS_STRING_MODE,
              e.QUOTE_STRING_MODE,
              un,
              e.C_BLOCK_COMMENT_MODE
            ]
          },
          e.C_LINE_COMMENT_MODE,
          e.C_BLOCK_COMMENT_MODE
        ]
      },
      un,
      s
    ]
  };
}
function nr(e) {
  const r = e.regex, o = {}, u = {
    begin: /\$\{/,
    end: /\}/,
    contains: [
      "self",
      {
        begin: /:-/,
        contains: [o]
      }
      // default values
    ]
  };
  Object.assign(o, {
    className: "variable",
    variants: [
      { begin: r.concat(
        /\$[\w\d#@][\w\d_]*/,
        // negative look-ahead tries to avoid matching patterns that are not
        // Perl at all like $ident$, @ident@, etc.
        "(?![\\w\\d])(?![$])"
      ) },
      u
    ]
  });
  const h = {
    className: "subst",
    begin: /\$\(/,
    end: /\)/,
    contains: [e.BACKSLASH_ESCAPE]
  }, n = {
    begin: /<<-?\s*(?=\w+)/,
    starts: { contains: [
      e.END_SAME_AS_BEGIN({
        begin: /(\w+)/,
        end: /(\w+)/,
        className: "string"
      })
    ] }
  }, t = {
    className: "string",
    begin: /"/,
    end: /"/,
    contains: [
      e.BACKSLASH_ESCAPE,
      o,
      h
    ]
  };
  h.contains.push(t);
  const i = {
    className: "",
    begin: /\\"/
  }, a = {
    className: "string",
    begin: /'/,
    end: /'/
  }, s = {
    begin: /\$?\(\(/,
    end: /\)\)/,
    contains: [
      {
        begin: /\d+#[0-9a-f]+/,
        className: "number"
      },
      e.NUMBER_MODE,
      o
    ]
  }, l = [
    "fish",
    "bash",
    "zsh",
    "sh",
    "csh",
    "ksh",
    "tcsh",
    "dash",
    "scsh"
  ], c = e.SHEBANG({
    binary: `(${l.join("|")})`,
    relevance: 10
  }), f = {
    className: "function",
    begin: /\w[\w\d_]*\s*\(\s*\)\s*\{/,
    returnBegin: !0,
    contains: [e.inherit(e.TITLE_MODE, { begin: /\w[\w\d_]*/ })],
    relevance: 0
  }, g = [
    "if",
    "then",
    "else",
    "elif",
    "fi",
    "for",
    "while",
    "in",
    "do",
    "done",
    "case",
    "esac",
    "function"
  ], d = [
    "true",
    "false"
  ], p = { match: /(\/[a-z._-]+)+/ }, b = [
    "break",
    "cd",
    "continue",
    "eval",
    "exec",
    "exit",
    "export",
    "getopts",
    "hash",
    "pwd",
    "readonly",
    "return",
    "shift",
    "test",
    "times",
    "trap",
    "umask",
    "unset"
  ], E = [
    "alias",
    "bind",
    "builtin",
    "caller",
    "command",
    "declare",
    "echo",
    "enable",
    "help",
    "let",
    "local",
    "logout",
    "mapfile",
    "printf",
    "read",
    "readarray",
    "source",
    "type",
    "typeset",
    "ulimit",
    "unalias"
  ], v = [
    "autoload",
    "bg",
    "bindkey",
    "bye",
    "cap",
    "chdir",
    "clone",
    "comparguments",
    "compcall",
    "compctl",
    "compdescribe",
    "compfiles",
    "compgroups",
    "compquote",
    "comptags",
    "comptry",
    "compvalues",
    "dirs",
    "disable",
    "disown",
    "echotc",
    "echoti",
    "emulate",
    "fc",
    "fg",
    "float",
    "functions",
    "getcap",
    "getln",
    "history",
    "integer",
    "jobs",
    "kill",
    "limit",
    "log",
    "noglob",
    "popd",
    "print",
    "pushd",
    "pushln",
    "rehash",
    "sched",
    "setcap",
    "setopt",
    "stat",
    "suspend",
    "ttyctl",
    "unfunction",
    "unhash",
    "unlimit",
    "unsetopt",
    "vared",
    "wait",
    "whence",
    "where",
    "which",
    "zcompile",
    "zformat",
    "zftp",
    "zle",
    "zmodload",
    "zparseopts",
    "zprof",
    "zpty",
    "zregexparse",
    "zsocket",
    "zstyle",
    "ztcp"
  ], w = [
    "chcon",
    "chgrp",
    "chown",
    "chmod",
    "cp",
    "dd",
    "df",
    "dir",
    "dircolors",
    "ln",
    "ls",
    "mkdir",
    "mkfifo",
    "mknod",
    "mktemp",
    "mv",
    "realpath",
    "rm",
    "rmdir",
    "shred",
    "sync",
    "touch",
    "truncate",
    "vdir",
    "b2sum",
    "base32",
    "base64",
    "cat",
    "cksum",
    "comm",
    "csplit",
    "cut",
    "expand",
    "fmt",
    "fold",
    "head",
    "join",
    "md5sum",
    "nl",
    "numfmt",
    "od",
    "paste",
    "ptx",
    "pr",
    "sha1sum",
    "sha224sum",
    "sha256sum",
    "sha384sum",
    "sha512sum",
    "shuf",
    "sort",
    "split",
    "sum",
    "tac",
    "tail",
    "tr",
    "tsort",
    "unexpand",
    "uniq",
    "wc",
    "arch",
    "basename",
    "chroot",
    "date",
    "dirname",
    "du",
    "echo",
    "env",
    "expr",
    "factor",
    // "false", // keyword literal already
    "groups",
    "hostid",
    "id",
    "link",
    "logname",
    "nice",
    "nohup",
    "nproc",
    "pathchk",
    "pinky",
    "printenv",
    "printf",
    "pwd",
    "readlink",
    "runcon",
    "seq",
    "sleep",
    "stat",
    "stdbuf",
    "stty",
    "tee",
    "test",
    "timeout",
    // "true", // keyword literal already
    "tty",
    "uname",
    "unlink",
    "uptime",
    "users",
    "who",
    "whoami",
    "yes"
  ];
  return {
    name: "Bash",
    aliases: ["sh"],
    keywords: {
      $pattern: /\b[a-z][a-z0-9._-]+\b/,
      keyword: g,
      literal: d,
      built_in: [
        ...b,
        ...E,
        // Shell modifiers
        "set",
        "shopt",
        ...v,
        ...w
      ]
    },
    contains: [
      c,
      // to catch known shells and boost relevancy
      e.SHEBANG(),
      // to catch unknown shells but still highlight the shebang
      f,
      s,
      e.HASH_COMMENT_MODE,
      n,
      p,
      t,
      i,
      a,
      o
    ]
  };
}
function tr(e) {
  const r = e.regex, o = e.COMMENT("--", "$"), u = {
    className: "string",
    variants: [
      {
        begin: /'/,
        end: /'/,
        contains: [{ begin: /''/ }]
      }
    ]
  }, h = {
    begin: /"/,
    end: /"/,
    contains: [{ begin: /""/ }]
  }, n = [
    "true",
    "false",
    // Not sure it's correct to call NULL literal, and clauses like IS [NOT] NULL look strange that way.
    // "null",
    "unknown"
  ], t = [
    "double precision",
    "large object",
    "with timezone",
    "without timezone"
  ], i = [
    "bigint",
    "binary",
    "blob",
    "boolean",
    "char",
    "character",
    "clob",
    "date",
    "dec",
    "decfloat",
    "decimal",
    "float",
    "int",
    "integer",
    "interval",
    "nchar",
    "nclob",
    "national",
    "numeric",
    "real",
    "row",
    "smallint",
    "time",
    "timestamp",
    "varchar",
    "varying",
    // modifier (character varying)
    "varbinary"
  ], a = [
    "add",
    "asc",
    "collation",
    "desc",
    "final",
    "first",
    "last",
    "view"
  ], s = [
    "abs",
    "acos",
    "all",
    "allocate",
    "alter",
    "and",
    "any",
    "are",
    "array",
    "array_agg",
    "array_max_cardinality",
    "as",
    "asensitive",
    "asin",
    "asymmetric",
    "at",
    "atan",
    "atomic",
    "authorization",
    "avg",
    "begin",
    "begin_frame",
    "begin_partition",
    "between",
    "bigint",
    "binary",
    "blob",
    "boolean",
    "both",
    "by",
    "call",
    "called",
    "cardinality",
    "cascaded",
    "case",
    "cast",
    "ceil",
    "ceiling",
    "char",
    "char_length",
    "character",
    "character_length",
    "check",
    "classifier",
    "clob",
    "close",
    "coalesce",
    "collate",
    "collect",
    "column",
    "commit",
    "condition",
    "connect",
    "constraint",
    "contains",
    "convert",
    "copy",
    "corr",
    "corresponding",
    "cos",
    "cosh",
    "count",
    "covar_pop",
    "covar_samp",
    "create",
    "cross",
    "cube",
    "cume_dist",
    "current",
    "current_catalog",
    "current_date",
    "current_default_transform_group",
    "current_path",
    "current_role",
    "current_row",
    "current_schema",
    "current_time",
    "current_timestamp",
    "current_path",
    "current_role",
    "current_transform_group_for_type",
    "current_user",
    "cursor",
    "cycle",
    "date",
    "day",
    "deallocate",
    "dec",
    "decimal",
    "decfloat",
    "declare",
    "default",
    "define",
    "delete",
    "dense_rank",
    "deref",
    "describe",
    "deterministic",
    "disconnect",
    "distinct",
    "double",
    "drop",
    "dynamic",
    "each",
    "element",
    "else",
    "empty",
    "end",
    "end_frame",
    "end_partition",
    "end-exec",
    "equals",
    "escape",
    "every",
    "except",
    "exec",
    "execute",
    "exists",
    "exp",
    "external",
    "extract",
    "false",
    "fetch",
    "filter",
    "first_value",
    "float",
    "floor",
    "for",
    "foreign",
    "frame_row",
    "free",
    "from",
    "full",
    "function",
    "fusion",
    "get",
    "global",
    "grant",
    "group",
    "grouping",
    "groups",
    "having",
    "hold",
    "hour",
    "identity",
    "in",
    "indicator",
    "initial",
    "inner",
    "inout",
    "insensitive",
    "insert",
    "int",
    "integer",
    "intersect",
    "intersection",
    "interval",
    "into",
    "is",
    "join",
    "json_array",
    "json_arrayagg",
    "json_exists",
    "json_object",
    "json_objectagg",
    "json_query",
    "json_table",
    "json_table_primitive",
    "json_value",
    "lag",
    "language",
    "large",
    "last_value",
    "lateral",
    "lead",
    "leading",
    "left",
    "like",
    "like_regex",
    "listagg",
    "ln",
    "local",
    "localtime",
    "localtimestamp",
    "log",
    "log10",
    "lower",
    "match",
    "match_number",
    "match_recognize",
    "matches",
    "max",
    "member",
    "merge",
    "method",
    "min",
    "minute",
    "mod",
    "modifies",
    "module",
    "month",
    "multiset",
    "national",
    "natural",
    "nchar",
    "nclob",
    "new",
    "no",
    "none",
    "normalize",
    "not",
    "nth_value",
    "ntile",
    "null",
    "nullif",
    "numeric",
    "octet_length",
    "occurrences_regex",
    "of",
    "offset",
    "old",
    "omit",
    "on",
    "one",
    "only",
    "open",
    "or",
    "order",
    "out",
    "outer",
    "over",
    "overlaps",
    "overlay",
    "parameter",
    "partition",
    "pattern",
    "per",
    "percent",
    "percent_rank",
    "percentile_cont",
    "percentile_disc",
    "period",
    "portion",
    "position",
    "position_regex",
    "power",
    "precedes",
    "precision",
    "prepare",
    "primary",
    "procedure",
    "ptf",
    "range",
    "rank",
    "reads",
    "real",
    "recursive",
    "ref",
    "references",
    "referencing",
    "regr_avgx",
    "regr_avgy",
    "regr_count",
    "regr_intercept",
    "regr_r2",
    "regr_slope",
    "regr_sxx",
    "regr_sxy",
    "regr_syy",
    "release",
    "result",
    "return",
    "returns",
    "revoke",
    "right",
    "rollback",
    "rollup",
    "row",
    "row_number",
    "rows",
    "running",
    "savepoint",
    "scope",
    "scroll",
    "search",
    "second",
    "seek",
    "select",
    "sensitive",
    "session_user",
    "set",
    "show",
    "similar",
    "sin",
    "sinh",
    "skip",
    "smallint",
    "some",
    "specific",
    "specifictype",
    "sql",
    "sqlexception",
    "sqlstate",
    "sqlwarning",
    "sqrt",
    "start",
    "static",
    "stddev_pop",
    "stddev_samp",
    "submultiset",
    "subset",
    "substring",
    "substring_regex",
    "succeeds",
    "sum",
    "symmetric",
    "system",
    "system_time",
    "system_user",
    "table",
    "tablesample",
    "tan",
    "tanh",
    "then",
    "time",
    "timestamp",
    "timezone_hour",
    "timezone_minute",
    "to",
    "trailing",
    "translate",
    "translate_regex",
    "translation",
    "treat",
    "trigger",
    "trim",
    "trim_array",
    "true",
    "truncate",
    "uescape",
    "union",
    "unique",
    "unknown",
    "unnest",
    "update",
    "upper",
    "user",
    "using",
    "value",
    "values",
    "value_of",
    "var_pop",
    "var_samp",
    "varbinary",
    "varchar",
    "varying",
    "versioning",
    "when",
    "whenever",
    "where",
    "width_bucket",
    "window",
    "with",
    "within",
    "without",
    "year"
  ], l = [
    "abs",
    "acos",
    "array_agg",
    "asin",
    "atan",
    "avg",
    "cast",
    "ceil",
    "ceiling",
    "coalesce",
    "corr",
    "cos",
    "cosh",
    "count",
    "covar_pop",
    "covar_samp",
    "cume_dist",
    "dense_rank",
    "deref",
    "element",
    "exp",
    "extract",
    "first_value",
    "floor",
    "json_array",
    "json_arrayagg",
    "json_exists",
    "json_object",
    "json_objectagg",
    "json_query",
    "json_table",
    "json_table_primitive",
    "json_value",
    "lag",
    "last_value",
    "lead",
    "listagg",
    "ln",
    "log",
    "log10",
    "lower",
    "max",
    "min",
    "mod",
    "nth_value",
    "ntile",
    "nullif",
    "percent_rank",
    "percentile_cont",
    "percentile_disc",
    "position",
    "position_regex",
    "power",
    "rank",
    "regr_avgx",
    "regr_avgy",
    "regr_count",
    "regr_intercept",
    "regr_r2",
    "regr_slope",
    "regr_sxx",
    "regr_sxy",
    "regr_syy",
    "row_number",
    "sin",
    "sinh",
    "sqrt",
    "stddev_pop",
    "stddev_samp",
    "substring",
    "substring_regex",
    "sum",
    "tan",
    "tanh",
    "translate",
    "translate_regex",
    "treat",
    "trim",
    "trim_array",
    "unnest",
    "upper",
    "value_of",
    "var_pop",
    "var_samp",
    "width_bucket"
  ], c = [
    "current_catalog",
    "current_date",
    "current_default_transform_group",
    "current_path",
    "current_role",
    "current_schema",
    "current_transform_group_for_type",
    "current_user",
    "session_user",
    "system_time",
    "system_user",
    "current_time",
    "localtime",
    "current_timestamp",
    "localtimestamp"
  ], f = [
    "create table",
    "insert into",
    "primary key",
    "foreign key",
    "not null",
    "alter table",
    "add constraint",
    "grouping sets",
    "on overflow",
    "character set",
    "respect nulls",
    "ignore nulls",
    "nulls first",
    "nulls last",
    "depth first",
    "breadth first"
  ], g = l, d = [
    ...s,
    ...a
  ].filter((w) => !l.includes(w)), p = {
    className: "variable",
    begin: /@[a-z0-9]+/
  }, b = {
    className: "operator",
    begin: /[-+*/=%^~]|&&?|\|\|?|!=?|<(?:=>?|<|>)?|>[>=]?/,
    relevance: 0
  }, E = {
    begin: r.concat(/\b/, r.either(...g), /\s*\(/),
    relevance: 0,
    keywords: { built_in: g }
  };
  function v(w, {
    exceptions: m,
    when: D
  } = {}) {
    const S = D;
    return m = m || [], w.map((A) => A.match(/\|\d+$/) || m.includes(A) ? A : S(A) ? `${A}|0` : A);
  }
  return {
    name: "SQL",
    case_insensitive: !0,
    // does not include {} or HTML tags `</`
    illegal: /[{}]|<\//,
    keywords: {
      $pattern: /\b[\w\.]+/,
      keyword: v(d, { when: (w) => w.length < 3 }),
      literal: n,
      type: i,
      built_in: c
    },
    contains: [
      {
        begin: r.either(...f),
        relevance: 0,
        keywords: {
          $pattern: /[\w\.]+/,
          keyword: d.concat(f),
          literal: n,
          type: i
        }
      },
      {
        className: "type",
        begin: r.either(...t)
      },
      E,
      p,
      u,
      h,
      e.C_NUMBER_MODE,
      e.C_BLOCK_COMMENT_MODE,
      o,
      b
    ]
  };
}
J.registerLanguage("xml", Kt);
J.registerLanguage("javascript", Xt);
J.registerLanguage("json", jt);
J.registerLanguage("yaml", Vt);
J.registerLanguage("plaintext", Jt);
J.registerLanguage("python", qt);
J.registerLanguage("java", er);
J.registerLanguage("bash", nr);
J.registerLanguage("sql", tr);
var N = /* @__PURE__ */ ((e) => (e.EQUAL = "equal", e.DELETE = "removed", e.ADD = "added", e.EMPTY = "empty", e))(N || {});
const ne = "<code-diff-modified>", te = "</code-diff-modified>", rr = ne.replace("<", "&lt;").replace(">", "&gt;"), ir = te.replace("<", "&lt;").replace(">", "&gt;");
function de(e) {
  return e === void 0 ? N.EQUAL : e.added ? N.ADD : e.removed ? N.DELETE : N.EQUAL;
}
function Ce(e, r, o = "word") {
  return typeof e == "undefined" ? r : typeof r == "undefined" ? e : (o === "char" ? Gn : Wn)(e, r).filter((h) => de(h) !== N.DELETE).map(
    (h) => de(h) === N.ADD ? `${ne}${h.value}${te}` : h.value
  ).join("");
}
function Sn(e, r) {
  const o = new Ue.diff_match_patch(), u = o.diff_linesToChars_(e, r), h = u.chars1, n = u.chars2, t = u.lineArray, i = o.diff_main(h, n, !1);
  return o.diff_charsToLines_(i, t), i.map((a) => {
    const [s, l] = a;
    return {
      count: l.trim().split(`
`).length,
      value: l,
      removed: s === Ue.DIFF_DELETE,
      added: s === Ue.DIFF_INSERT
    };
  });
}
function V(e, r) {
  if (!r.match(new RegExp(`(${ne}|${te})`, "g")))
    return J.highlight(r, { language: e }).value;
  let u = r;
  const h = r.replace(new RegExp(`(${ne}|${te})`, "g"), ""), n = document.createElement("div");
  n.innerHTML = J.highlight(h, { language: e }).value;
  let t = !1;
  const i = (a) => {
    a.childNodes.forEach((s) => {
      if (s.nodeType === Node.ELEMENT_NODE && i(s), s.nodeType === Node.TEXT_NODE) {
        if (!s.textContent)
          return;
        let l = s.textContent, c = "";
        for (t && (c = c + ne); l.length; ) {
          if (u.startsWith(ne)) {
            u = u.slice(ne.length), c = c + ne, t = !0;
            continue;
          }
          if (u.startsWith(te)) {
            u = u.slice(te.length), c = c + te, t = !1;
            continue;
          }
          const f = u.match(new RegExp(`(${ne}|${te})`)), g = f && f.index ? f.index : u.length, d = Math.min(g, l.length);
          c = c + u.substring(0, d), u = u.slice(d), l = l.slice(d);
        }
        t && (c = c + te), s.textContent = c;
      }
    });
  };
  return i(n), n.innerHTML.replace(new RegExp(rr, "g"), '<span class="x">').replace(new RegExp(ir, "g"), "</span>");
}
function Rn(e) {
  const r = (h, n) => (h.match(new RegExp(n, "g")) || []).length;
  let o = 0, u = 0;
  for (const h of e)
    h.added && (o += r(h.value.trim(), `
`) + 1), h.removed && (u += r(h.value.trim(), `
`) + 1);
  return { additionsNum: o, deletionsNum: u };
}
function sr(e, r, o = "plaintext", u = "word", h = 10) {
  const n = () => ({ type: N.EMPTY }), t = (p, b, E) => ({ type: p, num: b, code: E }), i = Sn(e, r);
  let a = 0, s = 0, l = !1;
  const c = [], f = {
    changes: c,
    collector: [],
    stat: Rn(i)
  };
  for (let p = 0; p < i.length; p++) {
    if (l) {
      l = !1;
      continue;
    }
    const [b, E] = [i[p], i[p + 1]], [v, w] = [de(b), de(E)], m = b.value.replace(/\n$/, "").split(`
`);
    if (E === void 0) {
      for (const S of m) {
        let A = n(), I = n();
        const R = V(o, S);
        v === N.EQUAL && (a++, s++, A = t(N.EQUAL, a, R), I = t(N.EQUAL, s, R)), v === N.DELETE && (a++, A = t(N.DELETE, a, R), I = n()), v === N.ADD && (s++, A = n(), I = t(N.ADD, s, R)), c.push({ left: A, right: I });
      }
      break;
    }
    if (v === N.EQUAL)
      for (const S of m) {
        a++, s++;
        const A = V(o, S);
        c.push({
          left: t(N.EQUAL, a, A),
          right: t(N.EQUAL, s, A)
        });
      }
    const D = E.value.replace(/\n$/, "").split(`
`);
    if (v === N.DELETE) {
      if (w === N.EQUAL)
        for (const S of m)
          a++, c.push({
            left: t(N.DELETE, a, V(o, S)),
            right: n()
          });
      if (w === N.ADD) {
        l = !0;
        const S = Math.max(b.count, E.count);
        for (let A = 0; A < S; A++) {
          A < b.count && a++, A < E.count && s++;
          const [I, R] = [m[A], D[A]], F = m.length === D.length ? Ce(R, I, u) : I, U = m.length === D.length ? Ce(I, R, u) : R, $ = A < b.count ? t(N.DELETE, a, V(o, F)) : n(), G = A < E.count ? t(N.ADD, s, V(o, U)) : n();
          c.push({ left: $, right: G });
        }
      }
    }
    if (v === N.ADD)
      for (const S of m)
        s++, c.push({
          left: n(),
          right: t(N.ADD, s, V(o, S))
        });
  }
  if (e === r) {
    for (let p = 0; p < c.length; p++)
      c[p].fold = !1;
    return f;
  }
  for (let p = 0; p < c.length; p++) {
    const b = c[p];
    if (b.left.type === N.DELETE || b.right.type === N.ADD) {
      const [E, v] = [Math.max(p - h, 0), Math.min(p + h + 1, c.length)];
      for (let w = E; w < v; w++)
        c[w].fold = !1;
    }
    b.fold === void 0 && (b.fold = !0);
  }
  const g = [];
  let d = [];
  for (let p = 0; p < c.length; p++) {
    const b = c[p];
    if (b.fold === !1) {
      d.length && (d[0].hideIndex = f.collector.length, f.collector.push({
        lines: d,
        fold: !0
      }), d = []), g.push(b);
      continue;
    }
    b.hide = !0, d.push(b), g.push(b);
  }
  return d.length && (d[0].hideIndex = f.collector.length, f.collector.push({
    lines: d,
    fold: !0
  }), d = []), f.changes = g, f;
}
function ar(e, r, o = "plaintext", u = "word", h = 10) {
  const n = Sn(e, r);
  let t = 0, i = 0, a = !1;
  const s = [], l = {
    changes: s,
    collector: [],
    stat: Rn(n)
  };
  for (let g = 0; g < n.length; g++) {
    if (a) {
      a = !1;
      continue;
    }
    const [d, p] = [n[g], n[g + 1]], [b, E] = [de(d), de(p)], v = d.value.replace(/\n$/, "").split(`
`);
    if (p === void 0) {
      for (const m of v) {
        b === N.EQUAL && (t++, i++), b === N.DELETE && t++, b === N.ADD && i++;
        const D = V(o, m);
        s.push({
          type: b,
          code: D,
          addNum: b === N.DELETE ? void 0 : i,
          delNum: b === N.ADD ? void 0 : t
        });
      }
      break;
    }
    if (b === N.EQUAL)
      for (const m of v) {
        t++, i++;
        const D = V(o, m);
        s.push({ type: N.EQUAL, code: D, delNum: t, addNum: i });
      }
    const w = p.value.replace(/\n$/, "").split(`
`);
    if (b === N.DELETE)
      if (E === N.ADD && v.length === w.length) {
        for (let m = 0; m < v.length; m++) {
          const D = v[m], S = w[m];
          t++;
          const A = V(o, Ce(S, D, u));
          s.push({ type: N.DELETE, code: A, delNum: t });
        }
        for (let m = 0; m < w.length; m++) {
          const D = v[m], S = w[m];
          i++;
          const A = V(o, Ce(D, S, u));
          s.push({ type: N.ADD, code: A, addNum: i });
        }
        a = !0;
      } else
        for (const m of v) {
          t++;
          const D = V(o, m);
          s.push({ type: N.DELETE, code: D, delNum: t });
        }
    if (b === N.ADD)
      for (const m of v) {
        i++;
        const D = V(o, m);
        s.push({ type: N.ADD, code: D, addNum: i });
      }
  }
  for (let g = 0; g < s.length; g++) {
    const d = s[g];
    if (d.type === N.DELETE || d.type === N.ADD) {
      const [p, b] = [Math.max(g - h, 0), Math.min(g + h + 1, s.length)];
      for (let E = p; E < b; E++)
        s[E].fold = !1;
    }
    d.fold === void 0 && (d.fold = !0);
  }
  if (e === r) {
    for (let g = 0; g < s.length; g++)
      s[g].fold = !1;
    return l;
  }
  const c = [];
  let f = [];
  for (let g = 0; g < s.length; g++) {
    const d = s[g];
    if (d.fold === !1) {
      f.length && (f[0].hideIndex = l.collector.length, l.collector.push({
        lines: f,
        fold: !0
      }), f = []), c.push(d);
      continue;
    }
    d.type === "equal" && (d.hide = !0, f.push(d)), c.push(d);
  }
  return f.length && (f[0].hideIndex = l.collector.length, l.collector.push({
    lines: f,
    fold: !0
  }), f = []), l.changes = c, l;
}
const or = { key: 0 }, lr = /* @__PURE__ */ P("td", {
  class: "blob-code blob-code-hunk",
  align: "left"
}, "  ", -1), cr = { key: 1 }, ur = ["data-code-marker", "innerHTML"], fr = /* @__PURE__ */ ye({
  __name: "UnifiedLine",
  props: {
    line: null
  },
  emits: ["expand"],
  setup(e, { emit: r }) {
    function o(u) {
      return u === N.DELETE ? "-" : u === N.ADD ? "+" : "";
    }
    return (u, h) => e.line.hideIndex !== void 0 && e.line.hide ? (W(), j("tr", or, [
      P("td", {
        class: "blob-num blob-num-hunk text-center",
        colspan: "2",
        onClick: h[0] || (h[0] = (n) => r("expand", e.line))
      }, " > "),
      lr
    ])) : e.line.hide ? fn("", !0) : (W(), j("tr", cr, [
      P("td", {
        class: be(["blob-num", {
          "blob-num-deletion": e.line.type === z(N).DELETE,
          "blob-num-addition": e.line.type === z(N).ADD,
          "blob-num-context": e.line.type === z(N).EQUAL,
          "blob-num-hunk": e.line.hide !== void 0
        }])
      }, he(e.line.delNum), 3),
      P("td", {
        class: be(["blob-num", {
          "blob-num-deletion": e.line.type === z(N).DELETE,
          "blob-num-addition": e.line.type === z(N).ADD,
          "blob-num-context": e.line.type === z(N).EQUAL,
          "blob-num-hunk": e.line.hide !== void 0
        }])
      }, he(e.line.addNum), 3),
      P("td", {
        class: be(["blob-code", {
          "blob-code-deletion": e.line.type === z(N).DELETE,
          "blob-code-addition": e.line.type === z(N).ADD,
          "blob-code-context": e.line.type === z(N).EQUAL,
          "blob-code-hunk": e.line.hide !== void 0
        }])
      }, [
        P("span", {
          class: "blob-code-inner blob-code-marker",
          "data-code-marker": o(e.line.type),
          innerHTML: e.line.code
        }, null, 8, ur)
      ], 2)
    ]));
  }
}), gr = { class: "diff-table" }, hr = /* @__PURE__ */ ye({
  __name: "UnifiedViewer",
  props: {
    diffChange: null
  },
  setup(e) {
    const r = e;
    function o({ hideIndex: u }) {
      u !== void 0 && r.diffChange.collector[u].lines.forEach((h) => {
        h.hide = !1, h.fold = !1;
      });
    }
    return (u, h) => {
      var n;
      return W(), j("table", gr, [
        P("tbody", null, [
          (W(!0), j(ge, null, Fe((n = e.diffChange) == null ? void 0 : n.changes, (t, i) => (W(), Oe(fr, {
            key: i,
            line: t,
            onExpand: o
          }, null, 8, ["line"]))), 128))
        ])
      ]);
    };
  }
}), dr = { key: 0 }, pr = /* @__PURE__ */ P("td", {
  class: "blob-code blob-code-inner blob-code-hunk",
  colspan: "3",
  align: "left"
}, "  ", -1), br = { key: 1 }, _r = /* @__PURE__ */ P("td", { class: "blob-num blob-num-empty empty-cell" }, null, -1), Er = /* @__PURE__ */ P("td", { class: "blob-code blob-code-empty empty-cell" }, null, -1), vr = ["data-code-marker", "innerHTML"], mr = /* @__PURE__ */ ye({
  __name: "SplitLine",
  props: {
    splitLine: null
  },
  emits: ["expand"],
  setup(e, { emit: r }) {
    function o(u) {
      return u === N.DELETE ? "-" : u === N.ADD ? "+" : "";
    }
    return (u, h) => e.splitLine.hideIndex !== void 0 && e.splitLine.hide ? (W(), j("tr", dr, [
      P("td", {
        class: "blob-num blob-num-hunk",
        colspan: "1",
        onClick: h[0] || (h[0] = (n) => r("expand", e.splitLine))
      }, " > "),
      pr
    ])) : e.splitLine.hide ? fn("", !0) : (W(), j("tr", br, [
      (W(!0), j(ge, null, Fe([e.splitLine.left, e.splitLine.right], (n) => (W(), j(ge, null, [
        n.type === z(N).EMPTY ? (W(), j(ge, { key: 0 }, [
          _r,
          Er
        ], 64)) : (W(), j(ge, { key: 1 }, [
          P("td", {
            class: be(["blob-num", {
              "blob-num-deletion": n.type === z(N).DELETE,
              "blob-num-addition": n.type === z(N).ADD,
              "blob-num-context": n.type === z(N).EQUAL,
              "blob-num-hunk": e.splitLine.hide !== void 0
            }])
          }, he(n.num), 3),
          P("td", {
            class: be(["blob-code", {
              "blob-code-deletion": n.type === z(N).DELETE,
              "blob-code-addition": n.type === z(N).ADD,
              "blob-code-context": n.type === z(N).EQUAL,
              "blob-code-hunk": e.splitLine.hide !== void 0
            }])
          }, [
            P("span", {
              class: "blob-code-inner blob-code-marker",
              "data-code-marker": o(n.type),
              innerHTML: n.code
            }, null, 8, vr)
          ], 2)
        ], 64))
      ], 64))), 256))
    ]));
  }
}), yr = { class: "file-diff-split diff-table" }, wr = /* @__PURE__ */ P("colgroup", null, [
  /* @__PURE__ */ P("col", { width: "44" }),
  /* @__PURE__ */ P("col"),
  /* @__PURE__ */ P("col", { width: "44" }),
  /* @__PURE__ */ P("col")
], -1), Nr = /* @__PURE__ */ ye({
  __name: "SplitViewer",
  props: {
    diffChange: null
  },
  setup(e) {
    const r = e;
    function o({ hideIndex: u }) {
      u !== void 0 && r.diffChange.collector[u].lines.forEach((h) => {
        h.hide = !1, h.fold = !1;
      });
    }
    return (u, h) => {
      var n;
      return W(), j("table", yr, [
        wr,
        P("tbody", null, [
          (W(!0), j(ge, null, Fe((n = e.diffChange) == null ? void 0 : n.changes, (t, i) => (W(), Oe(mr, {
            key: i,
            "split-line": t,
            onExpand: o
          }, null, 8, ["split-line"]))), 128))
        ])
      ]);
    };
  }
});
const Mr = { class: "file-header" }, Ar = { class: "file-info" }, Tr = { class: "filename" }, Dr = { class: "diff-stat" }, Sr = { class: "diff-stat-added" }, Rr = {
  class: "diff-stat-deleted",
  style: { "margin-left": "8px" }
}, Or = /* @__PURE__ */ ye({
  __name: "CodeDiff",
  props: {
    newString: null,
    oldString: null,
    language: { default: "plaintext" },
    context: { default: 10 },
    diffStyle: { default: "word" },
    outputFormat: { default: "line-by-line" },
    trim: { type: Boolean, default: !1 },
    noDiffLineFeed: { type: Boolean, default: !1 },
    maxHeight: { default: void 0 },
    filename: { default: void 0 }
  },
  emits: ["diff"],
  setup(e, { emit: r }) {
    const o = e, u = pe(() => o.outputFormat === "line-by-line"), h = pe(() => {
      let s = o.oldString || "";
      return s = o.trim ? s.trim() : s, s = o.noDiffLineFeed ? s.replace(/(\r\n)/g, `
`) : s, s;
    }), n = pe(() => {
      let s = o.newString || "";
      return s = o.trim ? s.trim() : s, s = o.noDiffLineFeed ? s.replace(/(\r\n)/g, `
`) : s, s;
    }), t = pe(
      () => u.value ? ar(h.value, n.value, o.language, o.diffStyle, o.context) : sr(h.value, n.value, o.language, o.diffStyle, o.context)
    ), i = Un(t.value), a = pe(() => i.value.stat.additionsNum === 0 && i.value.stat.deletionsNum === 0);
    return Pn(() => o, () => {
      i.value = t.value, r("diff", {
        stat: {
          isChanged: !a.value,
          addNum: i.value.stat.additionsNum,
          delNum: i.value.stat.deletionsNum
        }
      });
    }, { deep: !0, immediate: !0 }), (s, l) => (W(), j("div", {
      class: "code-diff-view",
      style: xn({ maxHeight: e.maxHeight })
    }, [
      P("div", Mr, [
        P("div", Ar, [
          P("span", Tr, he(e.filename), 1),
          P("span", Dr, [
            P("span", Sr, "+" + he(z(i).stat.additionsNum) + " additions", 1),
            P("span", Rr, "-" + he(z(i).stat.deletionsNum) + " deletions", 1)
          ])
        ])
      ]),
      z(u) ? (W(), Oe(hr, {
        key: 0,
        "diff-change": z(i)
      }, null, 8, ["diff-change"])) : (W(), Oe(Nr, {
        key: 1,
        "diff-change": z(i)
      }, null, 8, ["diff-change"]))
    ], 4));
  }
});
function Lr(e) {
  e.component("CodeDiff", Or);
}
const Ir = {
  install: Lr,
  hljs: J
};
export {
  Or as CodeDiff,
  Ir as default
};
